---
title: "Post hoc inference for differential gene expression studies"
author: "G. Blanchard, P. Neuvial and E. Roquain"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
bibliography: sansSouci.bib
vignette: >
  %\VignetteIndexEntry{Post hoc inference for differential gene expression studies}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
---
  
```{r setup, include = FALSE}
knitr::opts_chunk$set(
collapse = TRUE,
comment = "#>"
)
```

This vignette reproduces two figures from Section 8 in @BNR:chap. It demonstrates how the [`sansSouci`](https://github.com/pneuvial/sanssouci) package may be used to obtain post hoc confidence bounds on false positives in the case of differential gene expression analysis. More precisely, after showing the output of a classical differential analysis based on False Discovery Rate control, we focus on two applications of post hoc methods:

- confidence envelopes for the true or false positives
- statistical inference on volcano plots

```{r required-packages}
library("sansSouci")
library("ggplot2")
```

## Motivation: a differential gene expression study

We focus on differential gene expression studies in cancerology. These studies aim at identifying genes whose mean expression level differs significantly between two (or more) populations, based on a sample of gene expression measurements from individuals from these populations. Specifically, we consider a data set studied in @BGH2010.

```{r read-data}
filename <- "~/Downloads/bourgon.rds"
if (file.exists(filename)) {
  dat <- readRDS(filename)
} else {
  data_url <- url("https://plmbox.math.cnrs.fr/f/755496cc4c154a6dbab0/?dl=1")
  dat <- readRDS(data_url)
}
```

This data set which consists of gene expression measurements for  $n = `r ncol(dat)`$ patients with B-cell acute lymphoblastic leukemia (ALL) @CLGV+2005. These patients are classified into two subgoups, depending on whether or not they harbor a specific mutation called "BCR/ABL":

```{r colnames}
table(colnames(dat))
mut <- "BCR/ABL"
m <- nrow(dat)
```


The goal of this study is to understand the molecular differences at the gene expression level between the populations of BCR/ABL positive and negative ("NEG") patients. For each patient, we observe a vector of $m = `r m`$ gene expression values. 

The most basic question to ask is: 

> For which genes is there a difference in the mean expression level of the mutated and non-mutated population? 

This question can be addressed by performing one statistical test of no difference between means for each gene, and to define "differentially expressed" genes as those passing some significance threshold. One important concern here is the calibration of the significance threshold.

## Classical differential analysis

We start with a simple Welch test for differential expression for each gene. This can be done e.g. using the `sansSouci::rowWelchTests` function:

```{r row-welch-tests}
categ <- ifelse(colnames(dat) == "BCR/ABL", 1, 0) # map to 0/1
dex <- data.frame(rowWelchTests(dat, categ))
pval <- dex[["p.value"]]
```


We plot a histogram of the corresponding $p$-values:

```{r hist}
hist(pval, probability = TRUE, breaks = 20,
     xlab = "p-value", main = "p-value distributon")
```

As expected, the distribution presents a large number of small $p$-values (which include signals, i.e. differentially expressed genes) mixed with uniformly distributed $p$-values (corresponding to non-differentially expressed genes). 

### Multiple testing correction: False Discovery Rate control 

The state of the art approach to large-scale multiple testing is to control the False Discovery Rate (FDR), which is the expected proportion of wrongly selected genes (false positives) among all selected genes @benjamini95controlling. The most widely used method to control this risk is the Benjamini-Hochberg (BH) procedure, which has been shown to control the FDR when the hypotheses corresponding to the non-differentially expressed genes are independent @benjamini95controlling or satisfy a specific type of positive dependence called Positive Regression Dependence on the Subset $\mathcal{H}_0$ of truly non-differentially expressed genes @benjamini01control.

```{r bh}
q <- 0.05
adjp.BH <- p.adjust(pval, method = "BH")
nBH <- sum(adjp.BH <= q)
```

The application of the BH procedure at level $q = `r q`$ is illustrated in the figures below (all genes are displayed in the first one, second one is a zoom on the top genes):

```{r bh-plot}
my_col <- "#FF000040"
dex <- dex[order(pval), ]  ## order genes by increasing p-values
dex$gene_order <- 1:nrow(dex)

bh_plot <- ggplot(dex, aes(x = gene_order, y = p.value)) + 
  geom_line() +
  xlab("Number of top genes") + ylab("Ordered p-value") +
  geom_abline(slope = 1/m, intercept = 0, color = "#00000020", size = 2) +
  geom_abline(slope = q/m, color = my_col, size = 2) +
  # geom_segment(aes(x = nBH, y = 0, yend = q*nBH/m, xend = nBH), linetype = "dotted") +
  # geom_segment(aes(x = 0, y = q*nBH/m, xend = nBH, yend = q*nBH/m), linetype = "dotted") +
  geom_abline(slope = 0, intercept = q, linetype = "dotted", color = my_col, size = 2) +
  theme_bw() +
  theme(axis.text = element_text(size = 14), 
        axis.title = element_text(size = 18))  
#geom_text(x = 0, y = q, label = expression(alpha), color = my_col)

bh_plot
```

```{r bh-plot-zoom}
xmax <- nBH*2.5
bh_plot + 
  xlim(1, xmax) + ylim(0, dex$p.value[xmax]) +
  geom_segment(aes(x = nBH, y = 0, yend = q*nBH/m, xend = nBH), linetype = "dotted") +
  geom_segment(aes(x = 1, y = q*nBH/m, xend = nBH, yend = q*nBH/m), linetype = "dotted", col = 1)
```

## Confidence envelopes on "top-$k$" lists

In the absence of prior information on genes, a natural idea is to rank them by decreasing statistical significance, and a natural question to ask is: 

> Can we provide a lower confidence envelope on the number (or proportion) of truly differentially expressed genes among the "top $k$" genes, for a user-defined $k$?

We illustrate the use of post-hoc methods to provide this type of information. More specifcally, we build confidence statements on the number of true/false positives within the top $k$ most significant genes in a differential gene expression study, where $k$ may be defined by the user after seing the data, and multiple choices of $k$ are allowed. 

We start by defining a target confidence level:

```{r conf-level}
alpha <- 0.1
```

The method of @GS2011 makes it possible to calculate a confidence envelope on the number of true positives among the most significant genes. In the framework of @blanchard20post-hoc this bound is obtained from simes86improved inequality, by interpolation:

```{r plot-conf-envelope}
xmax <- 300
ce_Simes <- confidenceEnvelope(dex$p.value, refFamily = 'Simes', param = alpha)
plotConfidenceEnvelope(ce_Simes, xmax = xmax)
```

```{r TP_BH, echo=FALSE}
TP_BH <- subset(ce_Simes, x==nBH & stat == "TP")$bound
```

In view of the above plot, we can for example state that with probability larger than `r floor((1-alpha)*100)`, the number of false positives among the `r nBH` genes selected by the BH procedure is larger than `r TP_BH`.

### Adapting to unknown dependence yields tighter confidence bounds

However, as discussed in @blanchard20post-hoc, this bound has two major limitations, being a consequence of Simes' inequality:
- It is known to be valid only under certain positive dependence assumptions (PRDS) on the joint $p$-value distribution. Although the PRDS assumption is generally accepted in the case of differential expression studies (which in passing justifies the applicaiton of the BH procedure itself), it has not been formally proved to hold in this case.
- It is not adaptive to the specific type of dependence at hand for a particular data set.

To bypass these limitations, @blanchard20post-hoc have proposed a randomization-based procedure known as $\lambda$-calibration, which yields tighter bounds that are adapted to the dependency observed in the data set at hand. In the case of two-sample tests, this calibration is achieved by permutation of class labels:

```{r calibration}
cal <- calibrateJER(X = dat, categ, B = 100, alpha = alpha, refFamily = "Simes")
```

An alternative to the Simes reference family is the Beta reference family:

```{r calibration-beta}
cal_beta <- calibrateJER(X = dat, categ, B = 100, alpha = alpha, refFamily = "Beta", K = 50)
```


The obtained confidence envelopes can be compared graphically to the above Simes envelope:

```{r, conf-env-plot}
conf_envs <- list("Simes" = ce_Simes,
                  "Linear" = cal$conf_env,
                  "Beta (K=50)" = cal_beta$conf_env)
cols <- RColorBrewer::brewer.pal(3, "Dark2")
plotConfidenceEnvelope(conf_envs, xmax = 2000, cols = cols)
```

### Volcano plots

```{r volcano-params}
q <- 0.05
r <-  0.3
alpha <- 0.1
```

Let us assume that we are interested in genes selected by the BH procedure at level $q = `r q`$ and whose fold change is larger than $r = `r r`$ in absolute value. This corresponds to two sets of genes, with positive/negative fold change, which can be represented in the following plot:


```{r volcano-plot-Simes}
ylim <- c(0, 6)
thr <- SimesThresholdFamily(m)(alpha)
volcanoPlot(dat, categ = categ, thr = thr, q = q, r = r, ylim = ylim)
```

This type of plot is called a "volcano plot" @CC2003. Post hoc inference makes it possible to obtain statistical guarantees on selections such as the ones represented in the above figure. 

```{r volcano-plot-Simes-calibrated}
thr <- cal$thr
volcanoPlot(dat, categ = categ, thr = thr, q = q, r = r, ylim = ylim)
```

```{r volcano-plot-Beta-calibrated}
thr <- cal_beta$thr
volcanoPlot(dat, categ = categ, thr = thr, q = q, r = r, ylim = ylim)
```

## Session information

```{r}
sessionInfo()
```

## References


