<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Post hoc inference for differential gene expression studies • sansSouci</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Post hoc inference for differential gene expression studies">
<meta property="og:description" content="sansSouci">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sansSouci</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.8.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Simes_equi-correlation.html">Conservativeness of the Simes inequality under positive dependence</a>
    </li>
    <li>
      <a href="../articles/confidenceEnvelopes_localized.html">Confidence envelopes for spatially structured hypotheses</a>
    </li>
    <li>
      <a href="../articles/jointErrorRateCalibration_simulations.html">Joint Error Rate calibration</a>
    </li>
    <li>
      <a href="../articles/stepDownControl.html">Step-down joint Family-Wise Error Rate control</a>
    </li>
    <li>
      <a href="../articles/two-sample.html">Post hoc inference for differential gene expression studies</a>
    </li>
    <li>
      <a href="../articles/two-sample_golub.html">Post hoc inference for differential gene expression studies</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Post hoc inference for differential gene expression studies</h1>
                        <h4 class="author">G. Blanchard, P. Neuvial and E. Roquain</h4>
            
            <h4 class="date">2020-09-10</h4>
      
      
      <div class="hidden name"><code>two-sample.Rmd</code></div>

    </div>

    
    
<p>This vignette reproduces two figures from Section 8 in <span class="citation">Blanchard, Neuvial, and Roquain (2019)</span>. It demonstrates how the <a href="https://github.com/pneuvial/sanssouci"><code>sansSouci</code></a> package may be used to obtain post hoc confidence bounds on false positives in the case of differential gene expression analysis. More precisely, after showing the output of a classical differential analysis based on False Discovery Rate control, we focus on two applications of post hoc methods:</p>
<ul>
<li>confidence envelopes for the true or false positives</li>
<li>statistical inference on volcano plots</li>
</ul>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"sansSouci"</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'sansSouci'</span>
<span class="co">#&gt; The following object is masked from 'package:datasets':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     volcano</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"ggplot2"</span>)</pre></body></html></div>
<div id="motivation-a-differential-gene-expression-study" class="section level2">
<h2 class="hasAnchor">
<a href="#motivation-a-differential-gene-expression-study" class="anchor"></a>Motivation: a differential gene expression study</h2>
<p>We focus on differential gene expression studies in cancerology. These studies aim at identifying genes whose mean expression level differs significantly between two (or more) populations, based on a sample of gene expression measurements from individuals from these populations. Specifically, we consider a data set studied in <span class="citation">Bourgon, Gentleman, and Huber (2010)</span>.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">filename</span> <span class="kw">&lt;-</span> <span class="st">"~/Downloads/bourgon.rds"</span>
<span class="kw">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="no">filename</span>)) {
  <span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="no">filename</span>)
} <span class="kw">else</span> {
  <span class="no">data_url</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/connections.html">url</a></span>(<span class="st">"https://plmbox.math.cnrs.fr/f/755496cc4c154a6dbab0/?dl=1"</span>)
  <span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="no">data_url</span>)
}</pre></body></html></div>
<p>This data set which consists of gene expression measurements for <span class="math inline">\(n = 79\)</span> patients with B-cell acute lymphoblastic leukemia (ALL) <span class="citation">Chiaretti et al. (2005)</span>. These patients are classified into two subgoups, depending on whether or not they harbor a specific mutation called “BCR/ABL”:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">dat</span>))
<span class="co">#&gt; </span>
<span class="co">#&gt; BCR/ABL     NEG </span>
<span class="co">#&gt;      37      42</span>
<span class="no">mut</span> <span class="kw">&lt;-</span> <span class="st">"BCR/ABL"</span>
<span class="no">m</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">dat</span>)</pre></body></html></div>
<p>The goal of this study is to understand the molecular differences at the gene expression level between the populations of BCR/ABL positive and negative (“NEG”) patients. For each patient, we observe a vector of <span class="math inline">\(m = 12625\)</span> gene expression values.</p>
<p>The most basic question to ask is:</p>
<blockquote>
<p>For which genes is there a difference in the mean expression level of the mutated and non-mutated population?</p>
</blockquote>
<p>This question can be addressed by performing one statistical test of no difference between means for each gene, and to define “differentially expressed” genes as those passing some significance threshold. One important concern here is the calibration of the significance threshold.</p>
</div>
<div id="classical-differential-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#classical-differential-analysis" class="anchor"></a>Classical differential analysis</h2>
<p>We start with a simple Welch test for differential expression for each gene. This can be done e.g. using the <code><a href="../reference/rowWelchTests.html">sansSouci::rowWelchTests</a></code> function:</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">categ</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">dat</span>) <span class="kw">==</span> <span class="st">"BCR/ABL"</span>, <span class="fl">1</span>, <span class="fl">0</span>) <span class="co"># map to 0/1</span>
<span class="no">dex</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="fu"><a href="../reference/rowWelchTests.html">rowWelchTests</a></span>(<span class="no">dat</span>, <span class="no">categ</span>))
<span class="no">pval</span> <span class="kw">&lt;-</span> <span class="no">dex</span><span class="kw">[[</span><span class="st">"p.value"</span>]]</pre></body></html></div>
<p>We plot a histogram of the corresponding <span class="math inline">\(p\)</span>-values:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(<span class="no">pval</span>, <span class="kw">probability</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">breaks</span> <span class="kw">=</span> <span class="fl">20</span>,
     <span class="kw">xlab</span> <span class="kw">=</span> <span class="st">"p-value"</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"p-value distributon"</span>)</pre></body></html></div>
<p><img src="two-sample_files/figure-html/hist-1.png" width="700"></p>
<p>As expected, the distribution presents a large number of small <span class="math inline">\(p\)</span>-values (which include signals, i.e. differentially expressed genes) mixed with uniformly distributed <span class="math inline">\(p\)</span>-values (corresponding to non-differentially expressed genes).</p>
<div id="multiple-testing-correction-false-discovery-rate-control" class="section level3">
<h3 class="hasAnchor">
<a href="#multiple-testing-correction-false-discovery-rate-control" class="anchor"></a>Multiple testing correction: False Discovery Rate control</h3>
<p>The state of the art approach to large-scale multiple testing is to control the False Discovery Rate (FDR), which is the expected proportion of wrongly selected genes (false positives) among all selected genes <span class="citation">Benjamini and Hochberg (1995)</span>. The most widely used method to control this risk is the Benjamini-Hochberg (BH) procedure, which has been shown to control the FDR when the hypotheses corresponding to the non-differentially expressed genes are independent <span class="citation">Benjamini and Hochberg (1995)</span> or satisfy a specific type of positive dependence called Positive Regression Dependence on the Subset <span class="math inline">\(\mathcal{H}_0\)</span> of truly non-differentially expressed genes <span class="citation">Benjamini and Yekutieli (2001)</span>.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">q</span> <span class="kw">&lt;-</span> <span class="fl">0.05</span>
<span class="no">adjp.BH</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">pval</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">"BH"</span>)
<span class="no">nBH</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">adjp.BH</span> <span class="kw">&lt;=</span> <span class="no">q</span>)</pre></body></html></div>
<p>The application of the BH procedure at level <span class="math inline">\(q = 0.05\)</span> is illustrated in the figures below (all genes are displayed in the first one, second one is a zoom on the top genes):</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">my_col</span> <span class="kw">&lt;-</span> <span class="st">"#FF000040"</span>
<span class="no">dex</span> <span class="kw">&lt;-</span> <span class="no">dex</span>[<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">pval</span>), ]  <span class="co">## order genes by increasing p-values</span>
<span class="no">dex</span>$<span class="no">gene_order</span> <span class="kw">&lt;-</span> <span class="fl">1</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">dex</span>)

<span class="no">bh_plot</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">dex</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">gene_order</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">p.value</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"Number of top genes"</span>) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"Ordered p-value"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_abline</a></span>(<span class="kw">slope</span> <span class="kw">=</span> <span class="fl">1</span>/<span class="no">m</span>, <span class="kw">intercept</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"#00000020"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">2</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_abline</a></span>(<span class="kw">slope</span> <span class="kw">=</span> <span class="no">q</span>/<span class="no">m</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">my_col</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">2</span>) +
  <span class="co"># geom_segment(aes(x = nBH, y = 0, yend = q*nBH/m, xend = nBH), linetype = "dotted") +</span>
  <span class="co"># geom_segment(aes(x = 0, y = q*nBH/m, xend = nBH, yend = q*nBH/m), linetype = "dotted") +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_abline</a></span>(<span class="kw">slope</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">intercept</span> <span class="kw">=</span> <span class="no">q</span>, <span class="kw">linetype</span> <span class="kw">=</span> <span class="st">"dotted"</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">my_col</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">2</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">axis.text</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">14</span>),
        <span class="kw">axis.title</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">18</span>))
<span class="co">#geom_text(x = 0, y = q, label = expression(alpha), color = my_col)</span>

<span class="no">bh_plot</span></pre></body></html></div>
<p><img src="two-sample_files/figure-html/bh-plot-1.png" width="700"></p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">xmax</span> <span class="kw">&lt;-</span> <span class="no">nBH</span>*<span class="fl">2.5</span>
<span class="no">bh_plot</span> +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span>(<span class="fl">1</span>, <span class="no">xmax</span>) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span>(<span class="fl">0</span>, <span class="no">dex</span>$<span class="no">p.value</span>[<span class="no">xmax</span>]) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_segment.html">geom_segment</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">nBH</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">yend</span> <span class="kw">=</span> <span class="no">q</span>*<span class="no">nBH</span>/<span class="no">m</span>, <span class="kw">xend</span> <span class="kw">=</span> <span class="no">nBH</span>), <span class="kw">linetype</span> <span class="kw">=</span> <span class="st">"dotted"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_segment.html">geom_segment</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">q</span>*<span class="no">nBH</span>/<span class="no">m</span>, <span class="kw">xend</span> <span class="kw">=</span> <span class="no">nBH</span>, <span class="kw">yend</span> <span class="kw">=</span> <span class="no">q</span>*<span class="no">nBH</span>/<span class="no">m</span>), <span class="kw">linetype</span> <span class="kw">=</span> <span class="st">"dotted"</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="fl">1</span>)
<span class="co">#&gt; Warning: Removed 12218 row(s) containing missing values (geom_path).</span></pre></body></html></div>
<p><img src="two-sample_files/figure-html/bh-plot-zoom-1.png" width="700"></p>
</div>
</div>
<div id="confidence-envelopes-on-top-k-lists" class="section level2">
<h2 class="hasAnchor">
<a href="#confidence-envelopes-on-top-k-lists" class="anchor"></a>Confidence envelopes on “top-<span class="math inline">\(k\)</span>” lists</h2>
<p>In the absence of prior information on genes, a natural idea is to rank them by decreasing statistical significance, and a natural question to ask is:</p>
<blockquote>
<p>Can we provide a lower confidence envelope on the number (or proportion) of truly differentially expressed genes among the “top <span class="math inline">\(k\)</span>” genes, for a user-defined <span class="math inline">\(k\)</span>?</p>
</blockquote>
<p>We illustrate the use of post-hoc methods to provide this type of information. More specifcally, we build confidence statements on the number of true/false positives within the top <span class="math inline">\(k\)</span> most significant genes in a differential gene expression study, where <span class="math inline">\(k\)</span> may be defined by the user after seing the data, and multiple choices of <span class="math inline">\(k\)</span> are allowed.</p>
<p>We start by defining a target confidence level:</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">alpha</span> <span class="kw">&lt;-</span> <span class="fl">0.1</span></pre></body></html></div>
<p>The method of <span class="citation">Goeman and Solari (2011)</span> makes it possible to calculate a confidence envelope on the number of true positives among the most significant genes. In the framework of <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span> this bound is obtained from simes86improved inequality, by interpolation:</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">xmax</span> <span class="kw">&lt;-</span> <span class="fl">300</span>
<span class="no">ce_Simes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/confidenceEnvelope.html">confidenceEnvelope</a></span>(<span class="no">dex</span>$<span class="no">p.value</span>, <span class="kw">refFamily</span> <span class="kw">=</span> <span class="st">'Simes'</span>, <span class="kw">param</span> <span class="kw">=</span> <span class="no">alpha</span>)
<span class="fu"><a href="../reference/plotConfidenceEnvelope.html">plotConfidenceEnvelope</a></span>(<span class="no">ce_Simes</span>, <span class="kw">xmax</span> <span class="kw">=</span> <span class="no">xmax</span>)</pre></body></html></div>
<p><img src="two-sample_files/figure-html/plot-conf-envelope-1.png" width="700"></p>
<p>In view of the above plot, we can for example state that with probability larger than 90, the number of false positives among the 163 genes selected by the BH procedure is larger than 85.</p>
<div id="adapting-to-unknown-dependence-yields-tighter-confidence-bounds" class="section level3">
<h3 class="hasAnchor">
<a href="#adapting-to-unknown-dependence-yields-tighter-confidence-bounds" class="anchor"></a>Adapting to unknown dependence yields tighter confidence bounds</h3>
<p>However, as discussed in <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span>, this bound has two major limitations, being a consequence of Simes’ inequality: - It is known to be valid only under certain positive dependence assumptions (PRDS) on the joint <span class="math inline">\(p\)</span>-value distribution. Although the PRDS assumption is generally accepted in the case of differential expression studies (which in passing justifies the applicaiton of the BH procedure itself), it has not been formally proved to hold in this case. - It is not adaptive to the specific type of dependence at hand for a particular data set.</p>
<p>To bypass these limitations, <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span> have proposed a randomization-based procedure known as <span class="math inline">\(\lambda\)</span>-calibration, which yields tighter bounds that are adapted to the dependency observed in the data set at hand. In the case of two-sample tests, this calibration is achieved by permutation of class labels:</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">cal</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/calibrateJER.html">calibrateJER</a></span>(<span class="kw">X</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="no">categ</span>, <span class="kw">B</span> <span class="kw">=</span> <span class="fl">100</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="no">alpha</span>, <span class="kw">refFamily</span> <span class="kw">=</span> <span class="st">"Simes"</span>)</pre></body></html></div>
<p>An alternative to the Simes reference family is the Beta reference family:</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">cal_beta</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/calibrateJER.html">calibrateJER</a></span>(<span class="kw">X</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="no">categ</span>, <span class="kw">B</span> <span class="kw">=</span> <span class="fl">100</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="no">alpha</span>, <span class="kw">refFamily</span> <span class="kw">=</span> <span class="st">"Beta"</span>, <span class="kw">K</span> <span class="kw">=</span> <span class="fl">50</span>)</pre></body></html></div>
<p>The obtained confidence envelopes can be compared graphically to the above Simes envelope:</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">conf_envs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Simes"</span> <span class="kw">=</span> <span class="no">ce_Simes</span>,
                  <span class="st">"Linear"</span> <span class="kw">=</span> <span class="no">cal</span>$<span class="no">conf_env</span>,
                  <span class="st">"Beta (K=50)"</span> <span class="kw">=</span> <span class="no">cal_beta</span>$<span class="no">conf_env</span>)
<span class="no">cols</span> <span class="kw">&lt;-</span> <span class="kw pkg">RColorBrewer</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span>(<span class="fl">3</span>, <span class="st">"Dark2"</span>)
<span class="fu"><a href="../reference/plotConfidenceEnvelope.html">plotConfidenceEnvelope</a></span>(<span class="no">conf_envs</span>, <span class="kw">xmax</span> <span class="kw">=</span> <span class="fl">2000</span>, <span class="kw">cols</span> <span class="kw">=</span> <span class="no">cols</span>)</pre></body></html></div>
<p><img src="two-sample_files/figure-html/conf-env-plot-1.png" width="700"></p>
</div>
<div id="volcano-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#volcano-plots" class="anchor"></a>Volcano plots</h3>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">q</span> <span class="kw">&lt;-</span> <span class="fl">0.05</span>
<span class="no">r</span> <span class="kw">&lt;-</span>  <span class="fl">0.3</span>
<span class="no">alpha</span> <span class="kw">&lt;-</span> <span class="fl">0.1</span></pre></body></html></div>
<p>Let us assume that we are interested in genes selected by the BH procedure at level <span class="math inline">\(q = 0.05\)</span> and whose fold change is larger than <span class="math inline">\(r = 0.3\)</span> in absolute value. This corresponds to two sets of genes, with positive/negative fold change, which can be represented in the following plot:</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">ylim</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">6</span>)
<span class="no">thr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/SimesThresholdFamily.html">SimesThresholdFamily</a></span>(<span class="no">m</span>)(<span class="no">alpha</span>)
<span class="fu"><a href="../reference/volcanoPlot.html">volcanoPlot</a></span>(<span class="no">dat</span>, <span class="kw">categ</span> <span class="kw">=</span> <span class="no">categ</span>, <span class="kw">thr</span> <span class="kw">=</span> <span class="no">thr</span>, <span class="kw">q</span> <span class="kw">=</span> <span class="no">q</span>, <span class="kw">r</span> <span class="kw">=</span> <span class="no">r</span>, <span class="kw">ylim</span> <span class="kw">=</span> <span class="no">ylim</span>)</pre></body></html></div>
<p><img src="two-sample_files/figure-html/volcano-plot-Simes-1.png" width="700"></p>
<p>This type of plot is called a “volcano plot” <span class="citation">Cui and Churchill (2003)</span>. Post hoc inference makes it possible to obtain statistical guarantees on selections such as the ones represented in the above figure.</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">thr</span> <span class="kw">&lt;-</span> <span class="no">cal</span>$<span class="no">thr</span>
<span class="fu"><a href="../reference/volcanoPlot.html">volcanoPlot</a></span>(<span class="no">dat</span>, <span class="kw">categ</span> <span class="kw">=</span> <span class="no">categ</span>, <span class="kw">thr</span> <span class="kw">=</span> <span class="no">thr</span>, <span class="kw">q</span> <span class="kw">=</span> <span class="no">q</span>, <span class="kw">r</span> <span class="kw">=</span> <span class="no">r</span>, <span class="kw">ylim</span> <span class="kw">=</span> <span class="no">ylim</span>)</pre></body></html></div>
<p><img src="two-sample_files/figure-html/volcano-plot-Simes-calibrated-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="no">thr</span> <span class="kw">&lt;-</span> <span class="no">cal_beta</span>$<span class="no">thr</span>
<span class="fu"><a href="../reference/volcanoPlot.html">volcanoPlot</a></span>(<span class="no">dat</span>, <span class="kw">categ</span> <span class="kw">=</span> <span class="no">categ</span>, <span class="kw">thr</span> <span class="kw">=</span> <span class="no">thr</span>, <span class="kw">q</span> <span class="kw">=</span> <span class="no">q</span>, <span class="kw">r</span> <span class="kw">=</span> <span class="no">r</span>, <span class="kw">ylim</span> <span class="kw">=</span> <span class="no">ylim</span>)</pre></body></html></div>
<p><img src="two-sample_files/figure-html/volcano-plot-Beta-calibrated-1.png" width="700"></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R version 4.0.2 (2020-06-22)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Mojave 10.14.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/fr_FR.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] ggplot2_3.3.2   sansSouci_0.8.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.5         RColorBrewer_1.1-2 compiler_4.0.2     pillar_1.4.6      </span>
<span class="co">#&gt;  [5] tools_4.0.2        digest_0.6.25      evaluate_0.14      memoise_1.1.0     </span>
<span class="co">#&gt;  [9] lifecycle_0.2.0    tibble_3.0.3       gtable_0.3.0       lattice_0.20-41   </span>
<span class="co">#&gt; [13] pkgconfig_2.0.3    rlang_0.4.7        Matrix_1.2-18      rstudioapi_0.11   </span>
<span class="co">#&gt; [17] yaml_2.2.1         pkgdown_1.5.1      xfun_0.16          withr_2.2.0       </span>
<span class="co">#&gt; [21] stringr_1.4.0      dplyr_1.0.2        knitr_1.29         generics_0.0.2    </span>
<span class="co">#&gt; [25] desc_1.2.0         fs_1.5.0           vctrs_0.3.2        tidyselect_1.1.0  </span>
<span class="co">#&gt; [29] rprojroot_1.3-2    grid_4.0.2         glue_1.4.2         R6_2.4.1          </span>
<span class="co">#&gt; [33] rmarkdown_2.3      farver_2.0.3       purrr_0.3.4        magrittr_1.5      </span>
<span class="co">#&gt; [37] backports_1.1.9    scales_1.1.1       matrixStats_0.56.0 htmltools_0.5.0   </span>
<span class="co">#&gt; [41] ellipsis_0.3.1     MASS_7.3-52        assertthat_0.2.1   colorspace_1.4-1  </span>
<span class="co">#&gt; [45] labeling_0.3       stringi_1.4.6      munsell_0.5.0      crayon_1.3.4</span></pre></body></html></div>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-benjamini95controlling">
<p>Benjamini, Yoav, and Yosef Hochberg. 1995. “Controlling the False Discovery Rate: A Practical and Powerful Approach to Multiple Testing.” <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> 57 (1). Wiley Online Library: 289–300.</p>
</div>
<div id="ref-benjamini01control">
<p>Benjamini, Yoav, and Daniel Yekutieli. 2001. “The Control of the False Discovery Rate in Multiple Testing Under Dependency.” <em>Annals of Statistics</em>. JSTOR, 1165–88.</p>
</div>
<div id="ref-BNR:chap">
<p>Blanchard, Gilles, Pierre Neuvial, and Etienne Roquain. 2019. “On Agnostic Post Hoc Approaches to False Positive Control.” <a href="https://hal.archives-ouvertes.fr/hal-02320543">https://hal.archives-ouvertes.fr/hal-02320543</a>.</p>
</div>
<div id="ref-blanchard20post-hoc">
<p>———. 2020. “Post Hoc Confidence Bounds on False Positives Using Reference Families.” <em>Annals of Statistics</em> 48 (3): 1281–1303. <a href="https://projecteuclid.org/euclid.aos/1594972818">https://projecteuclid.org/euclid.aos/1594972818</a>.</p>
</div>
<div id="ref-BGH2010">
<p>Bourgon, Richard, Robert Gentleman, and Wolfgang Huber. 2010. “Independent filtering increases detection power for high-throughput experiments.” <em>PNAS</em>. <a href="https://doi.org/10.1073/pnas.0914005107/-/DCSupplemental.www.pnas.org/cgi/doi/10.1073/pnas.0914005107">https://doi.org/10.1073/pnas.0914005107/-/DCSupplemental.www.pnas.org/cgi/doi/10.1073/pnas.0914005107</a>.</p>
</div>
<div id="ref-CLGV+2005">
<p>Chiaretti, Sabina, Xiaochun Li, Robert Gentleman, Antonella Vitale, Kathy S Wang, Franco Mandelli, Robin Foa, and Jerome Ritz. 2005. “Gene Expression Profiles of B-Lineage Adult Acute Lymphocytic Leukemia Reveal Genetic Patterns That Identify Lineage Derivation and Distinct Mechanisms of Transformation.” <em>Clinical Cancer Research</em> 11 (20). AACR: 7209–19.</p>
</div>
<div id="ref-CC2003">
<p>Cui, Xiangqin, and Gary A Churchill. 2003. “Statistical Tests for Differential Expression in cDNA Microarray Experiments.” <em>Genome Biol</em> 4 (4): 210.</p>
</div>
<div id="ref-GS2011">
<p>Goeman, Jelle J, and Aldo Solari. 2011. “Multiple Testing for Exploratory Research.” <em>Statistical Science</em> 26 (4): 584–97.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gilles Blanchard, Guillermo Durand, <a href="https://www.math.univ-toulouse.fr/~pneuvial/">Pierre Neuvial</a>, <a href="https://etienneroquain-81.webself.net/">Etienne Roquain</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
