<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Confidence curves for structured hypotheses • sansSouci</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Confidence curves for structured hypotheses">
<meta property="og:description" content="sansSouci">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sansSouci</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.12.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/IIDEA.html">IIDEA: Interactive Inference for Differential Expression Analyses</a>
    </li>
    <li>
      <a href="../articles/Simes_equi-correlation.html">Conservativeness of the Simes inequality under positive dependence</a>
    </li>
    <li>
      <a href="../articles/confidenceCurves_localized.html">Confidence curves for structured hypotheses</a>
    </li>
    <li>
      <a href="../articles/jointErrorRateCalibration_simulations.html">Joint Error Rate calibration</a>
    </li>
    <li>
      <a href="../articles/post-hoc_differential-expression.html">Permutation-based post hoc inference for differential gene expression studies</a>
    </li>
    <li>
      <a href="../articles/post-hoc_differential-expression_RNAseq.html">Permutation-based post hoc inference for differential gene expression studies: RNAseq data</a>
    </li>
    <li>
      <a href="../articles/post-hoc_fMRI.html">Permutation-based post hoc inference for fMRI studies</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pneuvial/sanssouci/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="confidenceCurves_localized_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Confidence curves for structured hypotheses</h1>
                        <h4 class="author">G. Durand, G. Blanchard, P. Neuvial, E. Roquain</h4>
            
            <h4 class="date">2021-11-05</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pneuvial/sanssouci/blob/master/vignettes/confidenceCurves_localized.Rmd"><code>vignettes/confidenceCurves_localized.Rmd</code></a></small>
      <div class="hidden name"><code>confidenceCurves_localized.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The goal of this vignette is to illustrate the post hoc bounds on the number of true/false positives proposed in <span class="citation">Durand et al. (2020)</span> for localized signals. More specifically, we reproduce one of the plots of Figure 12 in <span class="citation">Durand et al. (2020)</span> using the R package <code>sansSouci</code>. We will explicitly quote <span class="citation">Durand et al. (2020)</span> when relevant.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/pneuvial/sanssouci">"sansSouci"</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="objective" class="section level2">
<h2 class="hasAnchor">
<a href="#objective" class="anchor"></a>Objective</h2>
<p>We consider <span class="math inline">\(m\)</span> null ordered hypotheses partitioned in intervals of size <span class="math inline">\(s\)</span>. For simplicity we set the number of intervals to be a power of 2: <span class="math inline">\(m = s 2^q\)</span> for some integer <span class="math inline">\(q\)</span>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">s</span> <span class="op">&lt;-</span> <span class="fl">100</span>
<span class="va">q</span> <span class="op">&lt;-</span> <span class="fl">7</span>
<span class="va">m</span> <span class="op">&lt;-</span> <span class="va">s</span><span class="op">*</span><span class="fl">2</span><span class="op">^</span><span class="va">q</span></code></pre></div>
<p>Therefore, we have <span class="math inline">\(m = 1.28\times 10^{4}\)</span>.Our goal is to compare three post hoc bounds. These bounds are obtained by interpolation from a <em>reference family</em> where the amount of signal is estimated by probabilistic inequalities, following the general principle laid down by <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span>, and they differ by the choice of the reference family:</p>
<ul>
<li><p>“Simes”: the bound derived from the <span class="citation">Simes (1986)</span> inequality as proposed by <span class="citation">Goeman and Solari (2011)</span> and further studied by <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span>. This bound was introduced in a context where the signal is not localized.</p></li>
<li><p>“Tree” and “Partition”: two bounds derived from the DKWM inequality (<span class="citation">Dvoretzky, Kiefer, and Wolfowitz (1956)</span>, <span class="citation">Massart (1990)</span>), as proposed in <span class="citation">Durand et al. (2020)</span>. For the “Partition” bound, the reference family is the original partition <span class="math inline">\((P_k)_k\)</span> of the <span class="math inline">\(m\)</span> null hypotheses into <span class="math inline">\(K=2^q\)</span> intervals. For the “Tree” bound, the reference family is the perfect binary tree whose leaves are the elements of the original partition.</p></li>
</ul>
</div>
<div id="settings" class="section level2">
<h2 class="hasAnchor">
<a href="#settings" class="anchor"></a>Settings</h2>
<p>We define the following numerical parameters, which characterize the true/false null hypothesis configuration considered in Section 5 of <span class="citation">Durand et al. (2020)</span>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">K1</span> <span class="op">&lt;-</span> <span class="fl">8</span>
<span class="va">r</span> <span class="op">&lt;-</span> <span class="fl">0.9</span>
<span class="va">m1</span> <span class="op">&lt;-</span>  <span class="va">r</span><span class="op">*</span><span class="va">K1</span><span class="op">*</span><span class="va">s</span>
<span class="va">barmu</span> <span class="op">&lt;-</span> <span class="fl">3</span></code></pre></div>
<p>More precisely, quoting <span class="citation">Durand et al. (2020)</span>:</p>
<blockquote>
<p>the false null hypotheses are contained in <span class="math inline">\(P_k\)</span> for <span class="math inline">\(1 \leq k \leq K_1\)</span>, for some fixed value of <span class="math inline">\(K_1\)</span>. The quantity <span class="math inline">\(r\)</span> is defined similarly as in (20), as the fraction of false null hypotheses in those <span class="math inline">\(P_k\)</span>, and is set to <span class="math inline">\(r =0.9\)</span>. All of the other partition pieces only contain true null hypotheses.</p>
</blockquote>
<p>We start by creating a binary tree structure and generating the signal:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SansSouciDyadic.html">SansSouciDyadic</a></span><span class="op">(</span><span class="va">m</span>, leaf_size <span class="op">=</span> <span class="va">s</span>, direction <span class="op">=</span> <span class="st">"top-down"</span><span class="op">)</span>
<span class="va">mu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gen.mu.leaves.html">gen.mu.leaves</a></span><span class="op">(</span>m <span class="op">=</span> <span class="va">m</span>, K1 <span class="op">=</span> <span class="va">K1</span>, d <span class="op">=</span> <span class="va">r</span>, grouped <span class="op">=</span> <span class="cn">TRUE</span>, setting <span class="op">=</span> <span class="st">"const"</span>, 
                    barmu <span class="op">=</span> <span class="va">barmu</span>, leaf_list <span class="op">=</span> <span class="va">obj</span><span class="op">$</span><span class="va">input</span><span class="op">$</span><span class="va">leaves</span><span class="op">)</span></code></pre></div>
<p>This construction is illustrated in the figure below (Figure 11 in <span class="citation">Durand et al. (2020)</span>) in the case where <span class="math inline">\(q=3\)</span>.</p>
<div class="figure">
<img src="DBNR_graph-simu.png" alt="Figure 11 in Durand et al." width="100%"><p class="caption">
Figure 11 in Durand et al.
</p>
</div>
<p>We generate <span class="math inline">\(p\)</span>-values according to the simulation setup in <span class="citation">Durand et al. (2020)</span>:</p>
<blockquote>
<p>The true null <span class="math inline">\(p\)</span>-values are distributed as i.i.d. <span class="math inline">\(\mathcal{N}(0,1)\)</span>, and false null <span class="math inline">\(p\)</span>-values are distributed as i.i.d. <span class="math inline">\(\mathcal{N}(\bar{\mu}, 1)\)</span>, where [<span class="math inline">\(\bar{\mu}= 3\)</span>].</p>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pvalues</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gen.p.values.html">gen.p.values</a></span><span class="op">(</span>m <span class="op">=</span> <span class="va">m</span>, mu <span class="op">=</span> <span class="va">mu</span>, rho <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
</div>
<div id="calculate-confidence-curves" class="section level2">
<h2 class="hasAnchor">
<a href="#calculate-confidence-curves" class="anchor"></a>Calculate confidence curves</h2>
<p>The confidence level for post hoc inference is set to <span class="math inline">\(\alpha = 0.05\)</span>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alpha</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></code></pre></div>
<blockquote>
<p>Below, we will be considering confidence curves of the form <span class="math inline">\((k, V(S_k))_{1 \leq k \leq m}\)</span>, where <span class="math inline">\(S_k\)</span> is the set of the <span class="math inline">\(k\)</span> smallest <span class="math inline">\(p\)</span>-values (regardless of the ordering given by the partition). Note that focusing on such sets is a priori favorable to the Simes bound, for which the elements of the reference family are among the <span class="math inline">\(S_k\)</span>.</p>
</blockquote>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pvalues</span><span class="op">)</span>
<span class="va">idxs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fu"><a href="../reference/all-generics.html">nHyp</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">)</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div id="true-number-of-false-positives" class="section level3">
<h3 class="hasAnchor">
<a href="#true-number-of-false-positives" class="anchor"></a>1- True number of false positives</h3>
<p>The true number of false positives will be called “Oracle” bound in the plots below.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">obj</span><span class="op">$</span><span class="va">input</span><span class="op">$</span><span class="va">truth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">mu</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">res_Oracle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">obj</span>, alpha <span class="op">=</span> <span class="va">alpha</span>, p.values <span class="op">=</span> <span class="va">pvalues</span>, family <span class="op">=</span> <span class="st">"Oracle"</span><span class="op">)</span>
<span class="va">res</span><span class="op">[[</span><span class="st">"Oracle"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">res_Oracle</span>, what <span class="op">=</span> <span class="st">"FP"</span>, all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">res</span><span class="op">[[</span><span class="st">"Oracle"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"Oracle"</span></code></pre></div>
</div>
<div id="simes-based-confidence-curve" class="section level3">
<h3 class="hasAnchor">
<a href="#simes-based-confidence-curve" class="anchor"></a>2- Simes-based confidence curve</h3>
<p>Here we use the <span class="citation">Simes (1986)</span> inequality to bound the number of false positives in each node of the tree, as proposed by <span class="citation">Goeman and Solari (2011)</span> and further studied by <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span>, both in a context where the signal is not localized.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_Simes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">obj</span>, alpha <span class="op">=</span> <span class="va">alpha</span>, p.values <span class="op">=</span> <span class="va">pvalues</span>, family <span class="op">=</span> <span class="st">"Simes"</span><span class="op">)</span>
<span class="va">res</span><span class="op">[[</span><span class="st">"Simes"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">res_Simes</span>, what <span class="op">=</span> <span class="st">"FP"</span>, all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">res</span><span class="op">[[</span><span class="st">"Simes"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"Simes"</span></code></pre></div>
</div>
<div id="dkwm-based-confidence-curve" class="section level3">
<h3 class="hasAnchor">
<a href="#dkwm-based-confidence-curve" class="anchor"></a>3- DKWM-based confidence curve</h3>
<p>Here we use the DKWM inequality (<span class="citation">Dvoretzky, Kiefer, and Wolfowitz (1956)</span>, <span class="citation">Massart (1990)</span>) to bound the number of false positives in each node of the tree, as suggested in <span class="citation">Durand et al. (2020)</span>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_DKWM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">obj</span>, <span class="va">alpha</span>, p.values <span class="op">=</span> <span class="va">pvalues</span>, family <span class="op">=</span> <span class="st">"DKWM"</span><span class="op">)</span>
<span class="va">res</span><span class="op">[[</span><span class="st">"Tree"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">res_DKWM</span>, what <span class="op">=</span> <span class="st">"FP"</span>, n_out <span class="op">=</span> <span class="fl">40</span><span class="op">)</span>
<span class="va">res</span><span class="op">[[</span><span class="st">"Tree"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"Tree"</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_DKWM_part</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">obj</span>, <span class="va">alpha</span>, p.values <span class="op">=</span> <span class="va">pvalues</span>, 
                family <span class="op">=</span> <span class="st">"DKWM"</span>, flavor <span class="op">=</span> <span class="st">"partition"</span><span class="op">)</span>
<span class="va">res</span><span class="op">[[</span><span class="st">"Partition"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">res_DKWM_part</span>, what <span class="op">=</span> <span class="st">"FP"</span>, n_out <span class="op">=</span> <span class="fl">40</span><span class="op">)</span>
<span class="va">res</span><span class="op">[[</span><span class="st">"Partition"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"Partition"</span></code></pre></div>
</div>
</div>
<div id="plot-confidence-curves" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-confidence-curves" class="anchor"></a>Plot confidence curves</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span>
<span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html">Reduce</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="va">res</span><span class="op">)</span>
<span class="va">lvls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Oracle"</span>, <span class="st">"Partition"</span>, <span class="st">"Simes"</span>, <span class="st">"Tree"</span>, <span class="st">"Hybrid"</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">lvls</span><span class="op">)</span>, <span class="st">"Set1"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">cols</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">lvls</span></code></pre></div>
<div id="upper-bound-on-the-number-of-false-positives" class="section level3">
<h3 class="hasAnchor">
<a href="#upper-bound-on-the-number-of-false-positives" class="anchor"></a>Upper bound on the number of false positives</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">xymax</span> <span class="op">&lt;-</span> <span class="va">m1</span>;
<span class="va">pV</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dat</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">bound</span>, colour <span class="op">=</span> <span class="va">method</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Upper bound on the number of false positives"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"sorted hypotheses"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">cols</span><span class="op">)</span>
<span class="va">pV</span></code></pre></div>
<p><img src="confidenceCurves_localized_files/figure-html/plot-confidence-curve-1.png" width="700"></p>
<p>The “Tree” and “Partition” bounds are sharper than the “Simes” bound as soon as we are considering “large” sets of hypotheses. The fact that the “Tree” and “Partition” bounds are not as sharp as the “Simes” bound for the first hundred of hypotheses can be explained by our choice of the ordering of the null hypotheses in the sets <span class="math inline">\(S_k\)</span>, which as discussed above is favorable to the “Simes” bound.</p>
<p>Zooming on the first 720 null hypotheses (in the order of the <span class="math inline">\(p\)</span>-values):</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pV</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">xymax</span><span class="op">)</span>,
                     ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">xymax</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="confidenceCurves_localized_files/figure-html/zoom-plot-confidence-curve-1.png" width="700"></p>
</div>
</div>
<div id="lower-bound-on-the-number-of-true-positives" class="section level2">
<h2 class="hasAnchor">
<a href="#lower-bound-on-the-number-of-true-positives" class="anchor"></a>Lower bound on the number of true positives</h2>
<p>The same information can be displayed as a lower bound on the number of true positives, defined for any <span class="math inline">\(S \subset \{1 \dots m\}\)</span> by <span class="math inline">\(|S| - V(S)\)</span>:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span><span class="op">$</span><span class="va">S</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">$</span><span class="va">x</span> <span class="op">-</span> <span class="va">dat</span><span class="op">$</span><span class="va">bound</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">xmax</span> <span class="op">&lt;-</span> <span class="va">m1</span>;
<span class="va">ymax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">S</span><span class="op">)</span>;
<span class="va">pS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dat</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">S</span>, colour <span class="op">=</span> <span class="va">method</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Lower bound on the number of true positives"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"sorted hypotheses"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">cols</span><span class="op">)</span>
<span class="va">pS</span></code></pre></div>
<p><img src="confidenceCurves_localized_files/figure-html/plot-lower-bound-1.png" width="700"></p>
<p>Zooming in the first 720 null hypotheses (in the order of the <span class="math inline">\(p\)</span>-values), we recover the middle plot in Figure 12 of <span class="citation">Durand et al. (2020)</span>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pS</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">xmax</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">ymax</span><span class="op">)</span></code></pre></div>
<p><img src="confidenceCurves_localized_files/figure-html/zoom-plot-lower-bound-1.png" width="700"></p>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ggplot2_3.3.5    sansSouci_0.12.4
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.7         RColorBrewer_1.1-2 pillar_1.6.4       compiler_4.1.2    
##  [5] jquerylib_0.1.4    highr_0.9          tools_4.1.2        digest_0.6.28     
##  [9] tibble_3.1.5       evaluate_0.14      memoise_2.0.0      lifecycle_1.0.1   
## [13] gtable_0.3.0       lattice_0.20-45    pkgconfig_2.0.3    rlang_0.4.12      
## [17] Matrix_1.3-4       yaml_2.2.1         pkgdown_1.6.1      xfun_0.28         
## [21] fastmap_1.1.0      withr_2.4.2        stringr_1.4.0      knitr_1.36        
## [25] vctrs_0.3.8        desc_1.4.0         generics_0.1.1     fs_1.5.0          
## [29] systemfonts_1.0.3  rprojroot_2.0.2    grid_4.1.2         glue_1.4.2        
## [33] R6_2.5.1           textshaping_0.3.6  fansi_0.5.0        rmarkdown_2.11    
## [37] farver_2.1.0       magrittr_2.0.1     ellipsis_0.3.2     scales_1.1.1      
## [41] htmltools_0.5.2    matrixStats_0.61.0 colorspace_2.0-2   labeling_0.4.2    
## [45] ragg_1.2.0         utf8_1.2.2         stringi_1.7.5      munsell_0.5.0     
## [49] cachem_1.0.6       crayon_1.4.2</code></pre>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-blanchard20post-hoc">
<p>Blanchard, Gilles, Pierre Neuvial, and Etienne Roquain. 2020. “Post Hoc Confidence Bounds on False Positives Using Reference Families.” <em>Annals of Statistics</em> 48 (3): 1281–1303. <a href="https://projecteuclid.org/euclid.aos/1594972818">https://projecteuclid.org/euclid.aos/1594972818</a>.</p>
</div>
<div id="ref-durand20post-hoc">
<p>Durand, Guillermo, Gilles Blanchard, Pierre Neuvial, and Etienne Roquain. 2020. “Post hoc false positive control for structured hypotheses.” <em>Scandinavian Journal of Statistics</em>. <a href="https://hal.archives-ouvertes.fr/hal-01829037">https://hal.archives-ouvertes.fr/hal-01829037</a>.</p>
</div>
<div id="ref-dvoretzky1956asymptotic">
<p>Dvoretzky, Aryeh, Jack Kiefer, and Jacob Wolfowitz. 1956. “Asymptotic Minimax Character of the Sample Distribution Function and of the Classical Multinomial Estimator.” <em>The Annals of Mathematical Statistics</em> 27 (3): 642–69.</p>
</div>
<div id="ref-GS2011">
<p>Goeman, Jelle J, and Aldo Solari. 2011. “Multiple Testing for Exploratory Research.” <em>Statistical Science</em> 26 (4): 584–97.</p>
</div>
<div id="ref-massart1990tight">
<p>Massart, Pascal. 1990. “The tight constant in the Dvoretzky-Kiefer-Wolfowitz inequality.” <em>The Annals of Probability</em> 18 (3): 1269–83.</p>
</div>
<div id="ref-simes86improved">
<p>Simes, R J. 1986. “An improved Bonferroni procedure for multiple tests of significance.” <em>Biometrika</em> 73 (3): 751–54.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.math.univ-toulouse.fr/~pneuvial/">Pierre Neuvial</a>, <a href="https://durandg12.github.io/">Guillermo Durand</a>, Nicolas Enjalbert-Courrech.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
