<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Permutation-based post hoc inference for fMRI studies • sansSouci</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Permutation-based post hoc inference for fMRI studies">
<meta property="og:description" content="sansSouci">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sansSouci</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.12.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/IIDEA.html">IIDEA: Interactive Inference for Differential Expression Analyses</a>
    </li>
    <li>
      <a href="../articles/Simes_equi-correlation.html">Conservativeness of the Simes inequality under positive dependence</a>
    </li>
    <li>
      <a href="../articles/confidenceCurves_localized.html">Confidence curves for structured hypotheses</a>
    </li>
    <li>
      <a href="../articles/jointErrorRateCalibration_simulations.html">Joint Error Rate calibration</a>
    </li>
    <li>
      <a href="../articles/post-hoc_differential-expression.html">Permutation-based post hoc inference for differential gene expression studies</a>
    </li>
    <li>
      <a href="../articles/post-hoc_differential-expression_RNAseq.html">Permutation-based post hoc inference for differential gene expression studies: RNAseq data</a>
    </li>
    <li>
      <a href="../articles/post-hoc_fMRI.html">Permutation-based post hoc inference for fMRI studies</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pneuvial/sanssouci/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="post-hoc_fMRI_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Permutation-based post hoc inference for fMRI studies</h1>
                        <h4 class="author">Alexandre Blain and Pierre Neuvial</h4>
            
            <h4 class="date">2021-11-05</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pneuvial/sanssouci/blob/master/vignettes/post-hoc_fMRI.Rmd"><code>vignettes/post-hoc_fMRI.Rmd</code></a></small>
      <div class="hidden name"><code>post-hoc_fMRI.Rmd</code></div>

    </div>

    
    
<p>This vignette shows how to use the <a href="https://github.com/pneuvial/sanssouci"><code>sansSouci</code></a> package to perform permutation-based post hoc inference for fMRI data. We focus on a motor task (left versus right click) from the Localizer data set by <span class="citation">Orfanos et al. (2017)</span>. The data set is included in the <a href="https://github.com/pneuvial/sanssouci.data"><code>sansSouci.data</code></a> package. It has been obtained using the Python module <a href="http://nilearn.github.io/">nilearn</a>, see <span class="citation">Abraham et al. (2014)</span>.</p>
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<p>We first make sure that the required R packages are installed and loaded.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span> <span class="op">||</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"ggplot2"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/pneuvial/sanssouci">"sansSouci"</a></span><span class="op">)</span> <span class="op">||</span> <span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"pneuvial/sanssouci@develop"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/pneuvial/sanssouci.data">"sansSouci.data"</a></span><span class="op">)</span> <span class="op">||</span> <span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"pneuvial/sanssouci.data"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/pneuvial/sanssouci">"sansSouci"</a></span><span class="op">)</span></code></pre></div>
<p>We set the seed of the random number generator for numerical reproducibility of the results:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">20200924</span><span class="op">)</span></code></pre></div>
</div>
<div id="loading-and-preprocessing-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-and-preprocessing-the-data" class="anchor"></a>Loading and preprocessing the data</h1>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">fMRI_localizer</span>, package <span class="op">=</span> <span class="st">"sansSouci.data"</span><span class="op">)</span>
<span class="va">Y</span> <span class="op">&lt;-</span> <span class="va">fMRI_localizer</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">fMRI_localizer</span><span class="op">)</span></code></pre></div>
<p>The samples are classified into two subgoups, These subgroups correspond to two different motor tasks the subjects were asked to perform, “left click” vs “right click”:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span>
<span class="va">categ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">categ</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">categ</th>
<th align="right">Freq</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">left</td>
<td align="right">15</td>
</tr>
<tr class="even">
<td align="left">right</td>
<td align="right">15</td>
</tr>
</tbody>
</table>
</div>
<div id="post-hoc-inference-by-the-simes-inequality" class="section level1">
<h1 class="hasAnchor">
<a href="#post-hoc-inference-by-the-simes-inequality" class="anchor"></a>Post hoc inference by the Simes inequality</h1>
<p>We start by calculating the Simes bound introduced by <span class="citation">Goeman and Solari (2011)</span>. This bound has already been applied to fMRI data, see <span class="citation">Rosenblatt et al. (2018)</span>. In that paper, post hoc inference (by the Simes inequality) is referred to as “All-resolutions inference” (ARI).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alpha</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></code></pre></div>
<p>We set the target confidence level to <span class="math inline">\(\alpha = 0.1\)</span>. The Simes bound is implemented in the package ‘cherry’. Here, we use the ‘sansSouci’ package, as follows:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span> <span class="op">==</span> <span class="st">"left"</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span> <span class="co"># map to 0/1</span>
<span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SansSouci.html">SansSouci</a></span><span class="op">(</span>Y <span class="op">=</span> <span class="va">Y</span>, groups <span class="op">=</span> <span class="va">groups</span><span class="op">)</span>
<span class="va">res_Simes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">obj</span>, B <span class="op">=</span> <span class="fl">0</span>, family <span class="op">=</span> <span class="st">"Simes"</span>, alpha <span class="op">=</span> <span class="va">alpha</span><span class="op">)</span> <span class="co">## B=0 =&gt; no calibration!</span></code></pre></div>
<p>Post hoc bounds are obtained from the <code>bound</code> function:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">TP_Simes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">res_Simes</span>, what <span class="op">=</span> <span class="st">"TP"</span><span class="op">)</span>
<span class="va">TP_Simes</span>
<span class="co">#&gt; [1] 88</span></code></pre></div>
<p>The Simes bound implies that with probability larger than 0.9, the number of voxels associated to the task of interest (among all 48900 voxels) is at least 88.</p>
<p>Disregarding (for now) the voxel locations in the brain, we plot confidence bounds on <span class="math inline">\(p\)</span>-value level sets (corresponding to maps obtained by thresholding at a given test statistic):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">xmax</span> <span class="op">&lt;-</span> <span class="fl">800</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">res_Simes</span>, xmax <span class="op">=</span> <span class="va">xmax</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Post hoc confidence curves: Simes/ARI"</span><span class="op">)</span></code></pre></div>
<p><img src="post-hoc_fMRI_files/figure-html/plot-conf-bounds-Simes-1.png" width="768"></p>
</div>
<div id="tighter-confidence-bounds-by-adaptation-to-unknown-dependence" class="section level1">
<h1 class="hasAnchor">
<a href="#tighter-confidence-bounds-by-adaptation-to-unknown-dependence" class="anchor"></a>Tighter confidence bounds by adaptation to unknown dependence</h1>
<p>As discussed in <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span>, the above-described bound is known to be valid only under certain positive dependence assumptions (PRDS) on the joint <span class="math inline">\(p\)</span>-value distribution. Although the PRDS assumption is widely accepted for fMRI studies (see <span class="citation">Genovese, Lazar, and Nichols (2002)</span>, <span class="citation">Nichols and Hayasaka (2003)</span>), we argue (and demonstrate below) that this assumption yields overly conservative post hoc bounds. Indeed, the Simes bound is by construction not <em>adaptive</em> to the specific type of dependence at hand for a particular data set.</p>
<p>To bypass these limitations, <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span> have proposed a randomization-based procedure known as <span class="math inline">\(\lambda\)</span>-calibration, which yields tighter bounds that are adapted to the dependency observed in the data set at hand. We note that a related approach has been proposed by <span class="citation">Hemerik, Solari, and Goeman (2019)</span>. In the case of two-sample tests, this calibration can be achieved by permutation of class labels, which is readily available via the <code>fit</code> function of the sansSouci package:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">B</span> <span class="op">&lt;-</span> <span class="fl">200</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">obj</span>, alpha <span class="op">=</span> <span class="va">alpha</span>, B <span class="op">=</span> <span class="va">B</span>, family <span class="op">=</span> <span class="st">"Simes"</span><span class="op">)</span></code></pre></div>
<p>An alternative to the Simes/Linear reference family is the Beta reference family:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">K</span> <span class="op">&lt;-</span> <span class="fl">500</span>
<span class="va">res_Beta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">res</span>, alpha <span class="op">=</span> <span class="va">alpha</span>, B <span class="op">=</span> <span class="va">B</span>, family <span class="op">=</span> <span class="st">"Beta"</span>, K <span class="op">=</span> <span class="va">K</span><span class="op">)</span></code></pre></div>
<p>As expected from the theory, the post hoc bounds obtained after calibration by these methods is much tighter than the Simes bound:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">resList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"Simes/ARI"</span> <span class="op">=</span> <span class="va">res_Simes</span>,
            <span class="st">"Linear"</span> <span class="op">=</span> <span class="va">res</span>,
            <span class="st">"Beta"</span> <span class="op">=</span> <span class="va">res_Beta</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">resList</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"Beta (K=%s)"</span>, <span class="va">K</span><span class="op">)</span>

<span class="va">bounds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">resList</span>, <span class="va">predict</span>, what <span class="op">=</span> <span class="st">"TP"</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">bounds</span>, digits <span class="op">=</span> <span class="fl">2</span>, col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Lower bound on True Positives"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Lower bound on True Positives</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Simes/ARI</td>
<td align="right">88</td>
</tr>
<tr class="even">
<td align="left">Linear</td>
<td align="right">321</td>
</tr>
<tr class="odd">
<td align="left">Beta (K=500)</td>
<td align="right">480</td>
</tr>
</tbody>
</table>
<p>In the next two sections we illustrate the use of these improved bounds in order to build</p>
<ul>
<li>confidence curves for the true or false positives</li>
<li>confidence statements for brain atlases</li>
</ul>
</div>
<div id="confidence-curves-on-top-k-lists" class="section level1">
<h1 class="hasAnchor">
<a href="#confidence-curves-on-top-k-lists" class="anchor"></a>Confidence curves on “top-<span class="math inline">\(k\)</span>” lists</h1>
<p>In the absence of prior information on voxels, a natural idea is to rank them by decreasing statistical significance, and a natural question to ask is:</p>
<blockquote>
<p>Can we provide a lower confidence curve on the number (or proportion) of truly associated voxels among the most significant ones?</p>
</blockquote>
<p>We illustrate the use of post-hoc methods to provide this type of information. More specifcally, we build confidence statements on the number of true/false positives within the top <span class="math inline">\(k\)</span> most significant voxels, where <span class="math inline">\(k\)</span> may be defined by the user after seing the data, and multiple choices of <span class="math inline">\(k\)</span> are allowed.</p>
<p>The confidence curves obtained by calibration of the Simes and Beta families can be compared graphically to the (parametric) Simes curve that can be obtained from <span class="citation">Goeman and Solari (2011)</span>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conf_bounds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">resList</span>, <span class="va">predict</span>, all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># cols &lt;- c("lightgray", "black", "darkgray")</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">conf_bounds</span><span class="op">)</span>, <span class="st">"Dark2"</span><span class="op">)</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotConfCurve.html">plotConfCurve</a></span><span class="op">(</span><span class="va">conf_bounds</span>, xmax <span class="op">=</span> <span class="fl">700</span>, cols <span class="op">=</span> <span class="va">cols</span><span class="op">)</span>
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">1.5</span><span class="op">)</span></code></pre></div>
<p><img src="post-hoc_fMRI_files/figure-html/plot-conf-bounds-1.png" width="768"></p>
<p>In this example, ARI is outperformed by permutation-based approaches, which are able to adapt to unknown dependency: for the same target risk level (<span class="math inline">\(\alpha = 0.1\)</span>), both permutation-based bounds are much less conservative than the classical Simes/ARI bound.</p>
</div>
<div id="post-hoc-bounds-on-brain-atlas-areas" class="section level1">
<h1 class="hasAnchor">
<a href="#post-hoc-bounds-on-brain-atlas-areas" class="anchor"></a>Post hoc bounds on brain atlas areas</h1>
<p>The goal in this section is to calculate post hoc bound on user-defined brain regions. One definition of such regions is given by the <a href="https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/Atlases">Harvard-Oxford brain atlas</a>, which is available from the sansSouci.data package.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">brainAtlas</span>, package <span class="op">=</span> <span class="st">"sansSouci.data"</span><span class="op">)</span></code></pre></div>
<p>This atlas is made of 48 areas. For each of these areas, we calculate the three post hoc bounds obtained above. Note that by construction of post hoc bounds, these 48 bounds are valid simultaneously (in particular, no further multiple testing correction is required at this stage).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bounds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">brainAtlas</span>, <span class="fl">1</span>, FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">reg</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">S</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">reg</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Region size"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">S</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">resList</span>, <span class="va">predict</span>, <span class="va">S</span>, <span class="st">"TP"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>In this particular case, some signal is revealed by the post hoc bound for 3 areas for permutation-based approaches:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ww</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="va">bounds</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">4</span>, <span class="op">]</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span>
<span class="va">cap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"Lower bounds on the number of true positives in %s brain regions."</span>,
               <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ww</span><span class="op">)</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">bounds</span><span class="op">[</span>, <span class="va">ww</span><span class="op">]</span><span class="op">)</span>, caption <span class="op">=</span> <span class="va">cap</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>Lower bounds on the number of true positives in 3 brain regions.</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="right">Region size</th>
<th align="right">Simes/ARI</th>
<th align="right">Linear</th>
<th align="right">Beta (K=500)</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Precentral Gyrus</td>
<td align="right">6640</td>
<td align="right">79</td>
<td align="right">242</td>
<td align="right">298</td>
</tr>
<tr class="even">
<td align="left">Postcentral Gyrus</td>
<td align="right">5030</td>
<td align="right">69</td>
<td align="right">194</td>
<td align="right">209</td>
</tr>
<tr class="odd">
<td align="left">Supramarginal Gyrus, anterior division</td>
<td align="right">2017</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Here again, ARI is outperformed by permutation-based approaches which are able to adapt to unknown dependency: for the same target risk level (<span class="math inline">\(\alpha = 0.1\)</span>) both permutation-based bounds are substantially less conservative than the classical Simes/ARI bound.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span>
<span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span>
<span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] sansSouci_0.12.4 ggplot2_3.3.5   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.7         RColorBrewer_1.1-2 highr_0.9          compiler_4.1.2    </span>
<span class="co">#&gt;  [5] pillar_1.6.4       tools_4.1.2        digest_0.6.28      lattice_0.20-45   </span>
<span class="co">#&gt;  [9] evaluate_0.14      memoise_2.0.0      lifecycle_1.0.1    tibble_3.1.5      </span>
<span class="co">#&gt; [13] gtable_0.3.0       pkgconfig_2.0.3    rlang_0.4.12       Matrix_1.3-4      </span>
<span class="co">#&gt; [17] yaml_2.2.1         pkgdown_1.6.1      xfun_0.28          fastmap_1.1.0     </span>
<span class="co">#&gt; [21] withr_2.4.2        stringr_1.4.0      knitr_1.36         generics_0.1.1    </span>
<span class="co">#&gt; [25] desc_1.4.0         fs_1.5.0           vctrs_0.3.8        systemfonts_1.0.3 </span>
<span class="co">#&gt; [29] rprojroot_2.0.2    grid_4.1.2         glue_1.4.2         R6_2.5.1          </span>
<span class="co">#&gt; [33] textshaping_0.3.6  fansi_0.5.0        rmarkdown_2.11     farver_2.1.0      </span>
<span class="co">#&gt; [37] magrittr_2.0.1     codetools_0.2-18   matrixStats_0.61.0 scales_1.1.1      </span>
<span class="co">#&gt; [41] htmltools_0.5.2    ellipsis_0.3.2     colorspace_2.0-2   labeling_0.4.2    </span>
<span class="co">#&gt; [45] ragg_1.2.0         utf8_1.2.2         stringi_1.7.5      munsell_0.5.0     </span>
<span class="co">#&gt; [49] cachem_1.0.6       crayon_1.4.2</span></code></pre></div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>To re-build this vignette from its source, use:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rmarkdown</span><span class="fu">::</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html">render</a></span><span class="op">(</span><span class="st">"post-hoc_fMRI.Rmd"</span>, output_format <span class="op">=</span> <span class="st">"pdf_document"</span><span class="op">)</span>
<span class="co"># To keep intermediate files, add option 'clean = FALSE'</span>
<span class="fu">rmarkdown</span><span class="fu">::</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html">render</a></span><span class="op">(</span><span class="st">"post-hoc_fMRI.Rmd"</span>, output_format <span class="op">=</span> <span class="st">"html_document"</span><span class="op">)</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-abraham2014machine-learning">
<p>Abraham, Alexandre, Fabian Pedregosa, Michael Eickenberg, Philippe Gervais, Andreas Mueller, Jean Kossaifi, Alexandre Gramfort, Bertrand Thirion, and Gaël Varoquaux. 2014. “Machine Learning for Neuroimaging with Scikit-Learn.” <em>Frontiers in Neuroinformatics</em> 8: 14.</p>
</div>
<div id="ref-blanchard20post-hoc">
<p>Blanchard, Gilles, Pierre Neuvial, and Etienne Roquain. 2020. “Post Hoc Confidence Bounds on False Positives Using Reference Families.” <em>Annals of Statistics</em> 48 (3): 1281–1303. <a href="https://projecteuclid.org/euclid.aos/1594972818">https://projecteuclid.org/euclid.aos/1594972818</a>.</p>
</div>
<div id="ref-genovese2002thresholding">
<p>Genovese, Christopher R, Nicole A Lazar, and Thomas Nichols. 2002. “Thresholding of Statistical Maps in Functional Neuroimaging Using the False Discovery Rate.” <em>Neuroimage</em> 15 (4): 870–78.</p>
</div>
<div id="ref-GS2011">
<p>Goeman, Jelle J, and Aldo Solari. 2011. “Multiple Testing for Exploratory Research.” <em>Statistical Science</em> 26 (4): 584–97.</p>
</div>
<div id="ref-HSG2019">
<p>Hemerik, Jesse, Aldo Solari, and Jelle J Goeman. 2019. “Permutation-Based Simultaneous Confidence Bounds for the False Discovery Proportion.” <em>Biometrika</em> 106 (3): 635–49.</p>
</div>
<div id="ref-nichols2003controlling">
<p>Nichols, Thomas, and Satoru Hayasaka. 2003. “Controlling the Familywise Error Rate in Functional Neuroimaging: A Comparative Review.” <em>Statistical Methods in Medical Research</em> 12 (5): 419–46.</p>
</div>
<div id="ref-orfanos2017brainomics">
<p>Orfanos, Dimitri Papadopoulos, Vincent Michel, Yannick Schwartz, Philippe Pinel, Antonio Moreno, Denis Le Bihan, and Vincent Frouin. 2017. “The Brainomics/Localizer Database.” <em>Neuroimage</em> 144: 309–14.</p>
</div>
<div id="ref-rosenblatt2018ari">
<p>Rosenblatt, Jonathan D, Livio Finos, Wouter D Weeda, Aldo Solari, and Jelle J Goeman. 2018. “All-Resolutions Inference for Brain Imaging.” <em>Neuroimage</em> 181: 786–96.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.math.univ-toulouse.fr/~pneuvial/">Pierre Neuvial</a>, <a href="https://durandg12.github.io/">Guillermo Durand</a>, Nicolas Enjalbert-Courrech, <a href="https://marie-perrotdockes.github.io/">Marie Perrot-Dockès</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
