<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Permutation-based post hoc inference for differential gene expression studies: RNAseq data • sansSouci</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Permutation-based post hoc inference for differential gene expression studies: RNAseq data">
<meta property="og:description" content="sansSouci">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sansSouci</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.12.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/IIDEA.html">IIDEA: Interactive Inference for Differential Expression Analyses</a>
    </li>
    <li>
      <a href="../articles/Simes_equi-correlation.html">Conservativeness of the Simes inequality under positive dependence</a>
    </li>
    <li>
      <a href="../articles/confidenceCurves_localized.html">Confidence curves for structured hypotheses</a>
    </li>
    <li>
      <a href="../articles/jointErrorRateCalibration_simulations.html">Joint Error Rate calibration</a>
    </li>
    <li>
      <a href="../articles/post-hoc_differential-expression.html">Permutation-based post hoc inference for differential gene expression studies</a>
    </li>
    <li>
      <a href="../articles/post-hoc_differential-expression_RNAseq.html">Permutation-based post hoc inference for differential gene expression studies: RNAseq data</a>
    </li>
    <li>
      <a href="../articles/post-hoc_fMRI.html">Permutation-based post hoc inference for fMRI studies</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pneuvial/sanssouci/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="post-hoc_differential-expression_RNAseq_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Permutation-based post hoc inference for differential gene expression studies: RNAseq data</h1>
                        <h4 class="author">Nicolas Enjalbert Courrech and Pierre Neuvial</h4>
            
            <h4 class="date">2021-10-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pneuvial/sanssouci/blob/master/vignettes/post-hoc_differential-expression_RNAseq.Rmd"><code>vignettes/post-hoc_differential-expression_RNAseq.Rmd</code></a></small>
      <div class="hidden name"><code>post-hoc_differential-expression_RNAseq.Rmd</code></div>

    </div>

    
    
<p>This vignette illustrate the relevance of permutation-based post hoc bounds on false positives for the differential analysis of (bulk) RNA sequencing (RNAseq) data. After showing the output of a classical differential analysis based on False Discovery Rate control we illustrate the application of basic post-hoc bounds derived from probabilistic inequalities. Then, we introduce more powerful post hoc methods (introduced by <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span>) that yield tighter bounds by adapting to unknown dependence by randomization. Finally we demonstrate the use of these methods on two applications of post hoc methods:</p>
<ul>
<li>confidence curves (envelopes) for the true or false positives</li>
<li>statistical inference on volcano plots</li>
</ul>
<p>The methods described in this vignette are described in detail in the book chapter <span class="citation">Blanchard, Neuvial, and Roquain (2019)</span> and in the paper <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span>. A shiny application for volcano plots is also available from <a href="https://shiny-iidea-sanssouci.apps.math.cnrs.fr/" class="uri">https://shiny-iidea-sanssouci.apps.math.cnrs.fr/</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span> <span class="op">||</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"ggplot2"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/pneuvial/sanssouci">"sansSouci"</a></span><span class="op">)</span> <span class="op">||</span> <span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"pneuvial/sanssouci@develop"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/pneuvial/sanssouci">"sansSouci"</a></span><span class="op">)</span></code></pre></div>
<p>Set the seed of the random number generator for numerical reproducibility of the results:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">20200924</span><span class="op">)</span></code></pre></div>
<div id="motivation-a-differential-gene-expression-study" class="section level1">
<h1 class="hasAnchor">
<a href="#motivation-a-differential-gene-expression-study" class="anchor"></a>Motivation: a differential gene expression study</h1>
<p>We focus on differential gene expression studies in cancerology. These studies aim at identifying genes whose mean expression level differs significantly between two (or more) populations, based on a sample of gene expression measurements from individuals from these populations.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"RNAseq_blca"</span>, package <span class="op">=</span> <span class="st">"sansSouci.data"</span><span class="op">)</span>
<span class="va">Y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="va">RNAseq_blca</span><span class="op">)</span>
<span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">RNAseq_blca</span><span class="op">)</span> <span class="op">==</span> <span class="st">"III"</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">RNAseq_blca</span><span class="op">)</span></code></pre></div>
<p>This data set consists of gene expression measurements for <span class="math inline">\(n = 270\)</span> patients from the Cancer Genome Atlas Urothelial Bladder Carcinoma (TCGA-BLCA) data collection (<span class="citation">Cancer Genome Atlas Research Network and others (2014)</span>). These patients are classified into two subgoups, depending on their clinical stage (II or III):</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">groups</span><span class="op">)</span>
<span class="co">#&gt; groups</span>
<span class="co">#&gt;   0   1 </span>
<span class="co">#&gt; 130 140</span>
<span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span></code></pre></div>
<p>The goal of this study is to understand the molecular differences at the gene expression level between stage II and stage III populations. For each patient, we observe a vector of <span class="math inline">\(m = 20196\)</span> gene expression values.</p>
<p>The most basic question to ask is:</p>
<blockquote>
<p>For which genes is there a difference in the mean expression level of the stage II and stage III populations?</p>
</blockquote>
<p>This question can be addressed by performing one statistical test per gene, and to define “differentially expressed” genes as those passing some significance threshold.</p>
<p>Below, we use the Wilcoxon (or Mann-Whitney) test. This is a non-parametric tests based on ranks. For each gene, we test the null hypothesis that the two populations have the same expression distribution against the alternative hypothesis that one of the two distribtions is stochastically greater than the other one. This can be done using the <code><a href="../reference/rowWilcoxonTests.html">sansSouci::rowWilcoxonTests</a></code> function:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="../reference/rowWilcoxonTests.html">rowWilcoxonTests</a></span><span class="op">(</span><span class="va">Y</span>, <span class="va">groups</span><span class="op">)</span><span class="op">)</span>
<span class="va">pval</span> <span class="op">&lt;-</span> <span class="va">dex</span><span class="op">[[</span><span class="st">"p.value"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p>We plot a histogram of the corresponding <span class="math inline">\(p\)</span>-values:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">pval</span>, probability <span class="op">=</span> <span class="cn">TRUE</span>, breaks <span class="op">=</span> <span class="fl">20</span>,
     xlab <span class="op">=</span> <span class="st">"p-value"</span>, main <span class="op">=</span> <span class="st">"p-value distributon"</span><span class="op">)</span></code></pre></div>
<p><img src="post-hoc_differential-expression_RNAseq_files/figure-html/hist-1.png" width="768"></p>
<p>As expected, the distribution presents a large number of small <span class="math inline">\(p\)</span>-values (which include signals, i.e. differentially expressed genes) mixed with uniformly distributed <span class="math inline">\(p\)</span>-values (corresponding to non-differentially expressed genes).</p>
<div id="multiple-testing-correction-false-discovery-rate-control" class="section level2">
<h2 class="hasAnchor">
<a href="#multiple-testing-correction-false-discovery-rate-control" class="anchor"></a>Multiple testing correction: False Discovery Rate control</h2>
<p>The state of the art approach to large-scale multiple testing is to control the False Discovery Rate (FDR), which is the expected proportion of wrongly selected genes (false positives) among all selected genes <span class="citation">Benjamini and Hochberg (1995)</span>. The most widely used method to control this risk is the Benjamini-Hochberg (BH) procedure, which has been shown to control the FDR when the hypotheses corresponding to the non-differentially expressed genes are independent <span class="citation">Benjamini and Hochberg (1995)</span> or satisfy a specific type of positive dependence called Positive Regression Dependence on the Subset (PRDS) <span class="math inline">\(\mathcal{H}_0\)</span> of truly non-differentially expressed genes <span class="citation">Benjamini and Yekutieli (2001)</span>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">q</span> <span class="op">&lt;-</span> <span class="fl">0.05</span>
<span class="va">adjp_BH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">pval</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>
<span class="va">dex</span><span class="op">$</span><span class="va">adjp</span> <span class="op">&lt;-</span> <span class="va">adjp_BH</span>
<span class="va">S_BH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">adjp_BH</span> <span class="op">&lt;=</span> <span class="va">q</span><span class="op">)</span>
<span class="va">nBH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">S_BH</span><span class="op">)</span>
<span class="va">nBH</span>
<span class="co">#&gt; [1] 2711</span></code></pre></div>
<p>The application of the BH procedure at level <span class="math inline">\(q = 0.05\)</span> is illustrated in the figures below (all genes are displayed in the first one, second one is a zoom on the top genes):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my_col</span> <span class="op">&lt;-</span> <span class="st">"#FF000080"</span>
<span class="va">dexo</span> <span class="op">&lt;-</span> <span class="va">dex</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>, <span class="op">]</span>  <span class="co">## order genes by increasing p-values</span>
<span class="va">dexo</span><span class="op">[[</span><span class="st">"gene_order"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">dex</span><span class="op">)</span>

<span class="va">bh_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dexo</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">gene_order</span>, y <span class="op">=</span> <span class="va">p.value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Number of top genes"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Ordered p-value"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="va">m</span>, intercept <span class="op">=</span> <span class="fl">0</span>, linetype <span class="op">=</span> <span class="fl">2</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="va">q</span><span class="op">/</span><span class="va">m</span>, color <span class="op">=</span> <span class="va">my_col</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="co"># geom_segment(aes(x = nBH, y = 0, yend = q*nBH/m, xend = nBH), linetype = "dotted") +</span>
  <span class="co"># geom_segment(aes(x = 0, y = q*nBH/m, xend = nBH, yend = q*nBH/m), linetype = "dotted") +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">0</span>, intercept <span class="op">=</span> <span class="va">q</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span>, color <span class="op">=</span> <span class="va">my_col</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>, 
        axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span><span class="op">)</span>  
<span class="co">#geom_text(x = 0, y = q, label = expression(alpha), color = my_col)</span>

<span class="va">bh_plot</span></code></pre></div>
<p><img src="post-hoc_differential-expression_RNAseq_files/figure-html/bh-plot-1.png" width="768"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">xmax</span> <span class="op">&lt;-</span> <span class="va">nBH</span><span class="op">*</span><span class="fl">2.5</span>
<span class="va">ymax</span> <span class="op">&lt;-</span> <span class="va">dexo</span><span class="op">$</span><span class="va">p.value</span><span class="op">[</span><span class="va">xmax</span><span class="op">]</span>
<span class="va">bh_plot</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">xmax</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">ymax</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_segment.html">geom_segment</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">nBH</span>, y <span class="op">=</span> <span class="fl">0</span>, yend <span class="op">=</span> <span class="va">q</span><span class="op">*</span><span class="va">nBH</span><span class="op">/</span><span class="va">m</span>, xend <span class="op">=</span> <span class="va">nBH</span><span class="op">)</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_segment.html">geom_segment</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1</span>, y <span class="op">=</span> <span class="va">q</span><span class="op">*</span><span class="va">nBH</span><span class="op">/</span><span class="va">m</span>, xend <span class="op">=</span> <span class="va">nBH</span>, yend <span class="op">=</span> <span class="va">q</span><span class="op">*</span><span class="va">nBH</span><span class="op">/</span><span class="va">m</span><span class="op">)</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span>, col <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 13419 row(s) containing missing values (geom_path).</span></code></pre></div>
<p><img src="post-hoc_differential-expression_RNAseq_files/figure-html/bh-plot-zoom-1.png" width="768"></p>
</div>
<div id="simes-posthoc-bounds" class="section level2">
<h2 class="hasAnchor">
<a href="#simes-posthoc-bounds" class="anchor"></a>Simes posthoc bounds</h2>
<p>Post hoc inference makes it possible to build confidence statements on the number of true/false positives within any set <span class="math inline">\(S\)</span> of genes: <span class="math inline">\(S\)</span> may be selected after seing the data (e.g., <span class="math inline">\(S\)</span> may be the set of rejections by the BH prcedure), and multiple choices of <span class="math inline">\(S\)</span> are allowed.</p>
<p>We begin by setting the target error level:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alpha</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></code></pre></div>
<p>Therefore, all of the statements made in this vignette are made at the 90% confidence level.</p>
<p>The Simes post hoc bound has been proposed by <span class="citation">Goeman and Solari (2011)</span>. It is valid under the same assumptions as the Benjamini-Hochberg method for FDR control (positive dependence between null test statistics, known as PRDS for Positive Regression Dependency on a Subset of hypotheses). In the framework of <span class="citation">Blanchard, Neuvial, and Roquain (2019)</span> this bound is a direct consequence of the <span class="citation">Simes (1986)</span> inequality. It can be applied to the 2711 rejections of the BH procedure as follows:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SansSouci.html">SansSouci</a></span><span class="op">(</span>Y <span class="op">=</span> <span class="va">Y</span>, groups <span class="op">=</span> <span class="va">groups</span><span class="op">)</span>
<span class="va">res_Simes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">obj</span>, B <span class="op">=</span> <span class="fl">0</span>, family <span class="op">=</span> <span class="st">"Simes"</span>, alpha <span class="op">=</span> <span class="va">alpha</span>, rowTestFUN <span class="op">=</span> <span class="va">rowWilcoxonTests</span><span class="op">)</span> <span class="co">## B=0 =&gt; no calibration!</span>
<span class="va">FP_Simes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">res_Simes</span>, <span class="va">S_BH</span>, what <span class="op">=</span> <span class="st">"FP"</span><span class="op">)</span></code></pre></div>
<p>The Simes bound implies that with probability larger than 0.9, the false discovery proportion among the genes selected by the BH procedure at level <span class="math inline">\(q= 0.05\)</span> is upper bounded by 0.46.</p>
</div>
</div>
<div id="tighter-confidence-bounds-by-adaptation-to-unknown-dependence" class="section level1">
<h1 class="hasAnchor">
<a href="#tighter-confidence-bounds-by-adaptation-to-unknown-dependence" class="anchor"></a>Tighter confidence bounds by adaptation to unknown dependence</h1>
<p>As discussed in <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span>, the above-described bound has two major limitations, being a consequence of the Simes inequality:</p>
<ul>
<li><p>It is known to be valid only under certain positive dependence assumptions (PRDS) on the joint <span class="math inline">\(p\)</span>-value distribution. Although the PRDS assumption is generally accepted in the case of differential expression studies (which justifies the application of the BH procedure itself), it has not been formally proved to hold in this case.</p></li>
<li><p>It is not <em>adaptive</em> to the specific type of dependence at hand for a particular data set.</p></li>
</ul>
<p>To bypass these limitations, <span class="citation">Blanchard, Neuvial, and Roquain (2020)</span> have proposed a randomization-based procedure known as <span class="math inline">\(\lambda\)</span>-calibration, which yields tighter bounds that are adapted to the dependency observed in the data set at hand. A closely related approach has been proposed by <span class="citation">Hemerik, Solari, and Goeman (2019)</span>. In the case of two-sample tests, this calibration can be achieved by permutation of class labels:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">B</span> <span class="op">&lt;-</span> <span class="fl">1000</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html">fit</a></span><span class="op">(</span><span class="va">obj</span>, B <span class="op">=</span> <span class="va">B</span>, alpha <span class="op">=</span> <span class="va">alpha</span>, family <span class="op">=</span> <span class="st">"Simes"</span>, rowTestFUN <span class="op">=</span> <span class="va">rowWilcoxonTests</span><span class="op">)</span></code></pre></div>
<p>As expected from the theory, the post hoc bounds obtained after calibration by these methods is much tighter than the Simes bound:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">resList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"Simes"</span> <span class="op">=</span> <span class="va">res_Simes</span>,
                <span class="st">"Simes + calibration"</span> <span class="op">=</span> <span class="va">res</span><span class="op">)</span>

<span class="va">bounds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">resList</span>, <span class="va">predict</span>, <span class="va">S_BH</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">bounds</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Lower bound on True Positives"</span>, <span class="st">"Upper bound on False Discovery Proportion"</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">bounds</span><span class="op">)</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Lower bound on True Positives</th>
<th align="right">Upper bound on False Discovery Proportion</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Simes</td>
<td align="right">1479</td>
<td align="right">0.45</td>
</tr>
<tr class="even">
<td align="left">Simes + calibration</td>
<td align="right">1950</td>
<td align="right">0.28</td>
</tr>
</tbody>
</table>
<p>In the next two sections we illustrate the use of these improved bounds in order to build</p>
<ul>
<li>confidence curves for the true or false positives</li>
<li>confidence statements for volcano plots</li>
</ul>
</div>
<div id="confidence-curves-on-top-k-lists" class="section level1">
<h1 class="hasAnchor">
<a href="#confidence-curves-on-top-k-lists" class="anchor"></a>Confidence curves on “top-<span class="math inline">\(k\)</span>” lists</h1>
<p>In the absence of prior information on genes, a natural idea is to rank them by decreasing statistical significance, and a natural question to ask is:</p>
<blockquote>
<p>Can we provide a lower confidence curve on the number (or proportion) of truly differentially expressed genes among the most significant genes?</p>
</blockquote>
<p>We illustrate the use of post-hoc methods to provide this type of information. More specifcally, we build confidence statements on the number of true/false positives within the top <span class="math inline">\(k\)</span> most significant genes in a differential gene expression study, where <span class="math inline">\(k\)</span> may be defined by the user after seing the data, and multiple choices of <span class="math inline">\(k\)</span> are allowed.</p>
<p>The confidence curves obtained by calibration of the Simes and Beta families can be compared graphically to the (parametric) Simes curve that can be obtained from <span class="citation">Goeman and Solari (2011)</span>:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conf_bounds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">resList</span>, <span class="va">predict</span>, all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"lightgray"</span><span class="op">)</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotConfCurve.html">plotConfCurve</a></span><span class="op">(</span><span class="va">conf_bounds</span>, xmax <span class="op">=</span> <span class="fl">5000</span>, cols <span class="op">=</span> <span class="va">cols</span><span class="op">)</span>
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">nBH</span>, color <span class="op">=</span> <span class="st">"gray"</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Wilcoxon Test"</span><span class="op">)</span></code></pre></div>
<p><img src="post-hoc_differential-expression_RNAseq_files/figure-html/conf-env-plot-1.png" width="768"></p>
<p>This example illustrates the power increase obtained by calibration.</p>
</div>
<div id="volcano-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#volcano-plots" class="anchor"></a>Volcano plots</h1>
<!-- For an interactive volcano plot, see the [volcano plot shiny application]( https://shiny-iidea-sanssouci.apps.math.cnrs.fr/). -->
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">q</span> <span class="op">&lt;-</span> <span class="fl">0.05</span>
<span class="va">r</span> <span class="op">&lt;-</span>  <span class="fl">0.25</span></code></pre></div>
<p>Let us assume that we are interested in genes selected by the BH procedure at level <span class="math inline">\(q = 0.05\)</span> and whose fold change is larger than <span class="math inline">\(r = 0.25\)</span> in absolute value. The “fold change” is an estimate of the effect size of a gene. Here, it is defined as the difference between the expression medians (on the log scale) of the two groups compared. This double selection by <span class="math inline">\(p\)</span>-value and fold change corresponds to two sets of genes, with positive/negative fold change, which can be represented in the following plot:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/volcanoPlot.html">volcanoPlot</a></span><span class="op">(</span><span class="va">res_Simes</span>, q <span class="op">=</span> <span class="va">q</span>, r <span class="op">=</span> <span class="va">r</span><span class="op">)</span></code></pre></div>
<p><img src="post-hoc_differential-expression_RNAseq_files/figure-html/volcano-Simes-1.png" width="768"></p>
<p>This type of plot is called a “volcano plot” <span class="citation">Cui and Churchill (2003)</span>. Post hoc inference makes it possible to obtain statistical guarantees on selections such as the ones represented in the above figure.</p>
<p>The substantial gain in power offered by the above-described calibration is illustrated as follows for the Simes reference family:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/volcanoPlot.html">volcanoPlot</a></span><span class="op">(</span><span class="va">res</span>, q <span class="op">=</span> <span class="va">q</span>, r <span class="op">=</span> <span class="va">r</span><span class="op">)</span></code></pre></div>
<p><img src="post-hoc_differential-expression_RNAseq_files/figure-html/volcano-Simes-cal-1.png" width="768"></p>
<p>The comparison between these bounds may be summarized by the following Table:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/all-generics.html">foldChanges</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span>
<span class="va">S_pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">fc</span> <span class="op">&gt;=</span> <span class="va">r</span> <span class="op">&amp;</span> <span class="va">adjp_BH</span> <span class="op">&lt;=</span> <span class="va">q</span><span class="op">)</span>
<span class="va">S_neg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">fc</span> <span class="op">&lt;=</span> <span class="op">-</span><span class="va">r</span> <span class="op">&amp;</span> <span class="va">adjp_BH</span> <span class="op">&lt;=</span> <span class="va">q</span><span class="op">)</span>
<span class="va">S_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">union</a></span><span class="op">(</span><span class="va">S_pos</span>, <span class="va">S_neg</span><span class="op">)</span>

<span class="va">all_bounds</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">S</span>, <span class="va">resList</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">S</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">resList</span>, <span class="va">predict</span>, <span class="va">S</span>, <span class="st">"TP"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="fu">all_bounds</span><span class="op">(</span><span class="va">S_pos</span>, <span class="va">resList</span><span class="op">)</span>, 
             <span class="fu">all_bounds</span><span class="op">(</span><span class="va">S_neg</span>, <span class="va">resList</span><span class="op">)</span>,
             <span class="fu">all_bounds</span><span class="op">(</span><span class="va">S_all</span>, <span class="va">resList</span><span class="op">)</span><span class="op">)</span>
<span class="va">plab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"BH-adjusted p.value &lt;"</span>, <span class="va">q</span><span class="op">)</span>
<span class="va">lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">plab</span>, <span class="st">"&amp;"</span>, <span class="st">" fold change  &gt; "</span>, <span class="va">r</span><span class="op">)</span>,
         <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">plab</span>, <span class="st">"&amp;"</span>, <span class="st">" fold change  &lt; "</span>, <span class="op">-</span><span class="va">r</span><span class="op">)</span>,
         <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">plab</span>, <span class="st">"&amp;"</span>, <span class="st">"|fold change| &gt; "</span>, <span class="va">r</span><span class="op">)</span><span class="op">)</span>
<span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">lab</span>, <span class="va">tab</span><span class="op">)</span>
<span class="va">cap</span> <span class="op">&lt;-</span> <span class="st">"Post hoc bounds on true positives in user-defined gene selections"</span>
<span class="co">#knitr::kable(tab, caption = cap, format = "latex")</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">tab</span>, caption <span class="op">=</span> <span class="va">cap</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>Post hoc bounds on true positives in user-defined gene selections</caption>
<thead><tr class="header">
<th align="left">lab</th>
<th align="left"></th>
<th align="left">Simes</th>
<th align="left">Simes + calibration</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">BH-adjusted p.value &lt; 0.05 &amp; fold change &gt; 0.25</td>
<td align="left">1525</td>
<td align="left">828</td>
<td align="left">969</td>
</tr>
<tr class="even">
<td align="left">BH-adjusted p.value &lt; 0.05 &amp; fold change &lt; -0.25</td>
<td align="left">526</td>
<td align="left">56</td>
<td align="left">124</td>
</tr>
<tr class="odd">
<td align="left">BH-adjusted p.value &lt; 0.05 &amp; |fold change| &gt; 0.25</td>
<td align="left">2051</td>
<td align="left">1118</td>
<td align="left">1366</td>
</tr>
</tbody>
</table>
</div>
<div id="volcano-plot-using-limma-voom-statistics" class="section level1">
<h1 class="hasAnchor">
<a href="#volcano-plot-using-limma-voom-statistics" class="anchor"></a>Volcano plot using limma-voom statistics</h1>
<p>Post hoc bounds can be calculated for any gene selection. In particular, even if Wilcoxon tests have been performed <em>for the calibration of the post hoc bounds</em>, it is possible to rely on other statistics to <em>select genes of interest</em>. In this section, we illustrate this idea by making a volcano plot based on the <span class="math inline">\(p\)</span>-values and log-fold changes obtained from the limma-voom method of <span class="citation">Law et al. (2014)</span> implemented in the <code>limma</code> package (<span class="citation">Ritchie et al. (2015)</span>):</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma">limma</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span>
<span class="va">X</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">exp</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span>  <span class="co"># back to linear scale</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span><span class="op">(</span><span class="va">X</span><span class="op">)</span>
<span class="co">#&gt; Repeated column names found in count matrix</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span>
<span class="va">Grp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">groups</span><span class="op">)</span>
<span class="va">mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="fl">0</span> <span class="op">+</span> <span class="va">Grp</span><span class="op">)</span>

<span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html">voom</a></span><span class="op">(</span><span class="va">d</span>, <span class="va">mm</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">res_lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">mm</span><span class="op">)</span>
<span class="va">contr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html">makeContrasts</a></span><span class="op">(</span><span class="va">Grp1</span> <span class="op">-</span> <span class="va">Grp0</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">res_lm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">res_fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html">contrasts.fit</a></span><span class="op">(</span><span class="va">res_lm</span>, <span class="va">contr</span><span class="op">)</span>
<span class="va">res_eb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">res_fit</span><span class="op">)</span>
<span class="va">TT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="va">res_eb</span>, sort.by <span class="op">=</span> <span class="st">"none"</span>, number <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></code></pre></div>
<p>The next plot that the <span class="math inline">\(p\)</span>-values obtained from limma-voom and from Wilcoxon tests are consistent.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>wilcox <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>, limma <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">TT</span><span class="op">$</span><span class="va">P.Value</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">limma</span>, y <span class="op">=</span> <span class="va">wilcox</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#10101010"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"p-values (log-scale)"</span><span class="op">)</span></code></pre></div>
<p><img src="post-hoc_differential-expression_RNAseq_files/figure-html/compare-p-values-1.png" width="768"></p>
<p>Finally we obtain the volcano plot based on limma statistics.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="fu"><a href="../reference/volcanoPlot.html">volcanoPlot</a></span><span class="op">(</span><span class="va">res_Simes</span>, 
            fold_changes <span class="op">=</span> <span class="va">TT</span><span class="op">$</span><span class="va">logFC</span>, p_values <span class="op">=</span> <span class="va">TT</span><span class="op">$</span><span class="va">P.Value</span>, 
            q <span class="op">=</span> <span class="va">q</span>, r <span class="op">=</span> <span class="va">r</span><span class="op">)</span></code></pre></div>
<p><img src="post-hoc_differential-expression_RNAseq_files/figure-html/volcano-limma-1.png" width="768"></p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.1 (2021-08-10)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span>
<span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span>
<span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] edgeR_3.34.1     limma_3.48.3     sansSouci_0.12.4 ggplot2_3.3.5   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.7         highr_0.9          compiler_4.1.1     pillar_1.6.4      </span>
<span class="co">#&gt;  [5] tools_4.1.1        digest_0.6.28      lattice_0.20-45    evaluate_0.14     </span>
<span class="co">#&gt;  [9] memoise_2.0.0      lifecycle_1.0.1    tibble_3.1.5       gtable_0.3.0      </span>
<span class="co">#&gt; [13] pkgconfig_2.0.3    rlang_0.4.12       Matrix_1.3-4       yaml_2.2.1        </span>
<span class="co">#&gt; [17] pkgdown_1.6.1      xfun_0.27          fastmap_1.1.0      withr_2.4.2       </span>
<span class="co">#&gt; [21] stringr_1.4.0      knitr_1.36         generics_0.1.0     desc_1.4.0        </span>
<span class="co">#&gt; [25] fs_1.5.0           vctrs_0.3.8        systemfonts_1.0.3  locfit_1.5-9.4    </span>
<span class="co">#&gt; [29] rprojroot_2.0.2    grid_4.1.1         glue_1.4.2         R6_2.5.1          </span>
<span class="co">#&gt; [33] textshaping_0.3.6  fansi_0.5.0        rmarkdown_2.11     farver_2.1.0      </span>
<span class="co">#&gt; [37] magrittr_2.0.1     codetools_0.2-18   matrixStats_0.61.0 scales_1.1.1      </span>
<span class="co">#&gt; [41] htmltools_0.5.2    ellipsis_0.3.2     colorspace_2.0-2   labeling_0.4.2    </span>
<span class="co">#&gt; [45] ragg_1.1.3         utf8_1.2.2         stringi_1.7.5      munsell_0.5.0     </span>
<span class="co">#&gt; [49] cachem_1.0.6       crayon_1.4.1</span></code></pre></div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>To re-build this vignette from its source, use:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rmarkdown</span><span class="fu">::</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html">render</a></span><span class="op">(</span><span class="st">"post-hoc_differential-expression_RNAseq.Rmd"</span>, output_format <span class="op">=</span> <span class="st">"pdf_document"</span><span class="op">)</span>
<span class="co"># To keep intermediate files, add option 'clean = FALSE'</span>
<span class="fu">rmarkdown</span><span class="fu">::</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html">render</a></span><span class="op">(</span><span class="st">"post-hoc_differential-expression_RNAseq.Rmd"</span>, output_format <span class="op">=</span> <span class="st">"html_document"</span><span class="op">)</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-benjamini95controlling">
<p>Benjamini, Yoav, and Yosef Hochberg. 1995. “Controlling the False Discovery Rate: A Practical and Powerful Approach to Multiple Testing.” <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> 57 (1): 289–300.</p>
</div>
<div id="ref-benjamini01control">
<p>Benjamini, Yoav, and Daniel Yekutieli. 2001. “The Control of the False Discovery Rate in Multiple Testing Under Dependency.” <em>Annals of Statistics</em>, 1165–88.</p>
</div>
<div id="ref-blanchard20post-hoc">
<p>Blanchard, Gilles, Pierre Neuvial, and Etienne Roquain. 2020. “Post Hoc Confidence Bounds on False Positives Using Reference Families.” <em>Annals of Statistics</em> 48 (3): 1281–1303. <a href="https://projecteuclid.org/euclid.aos/1594972818">https://projecteuclid.org/euclid.aos/1594972818</a>.</p>
</div>
<div id="ref-BNR:chap">
<p>———. 2019. “On Agnostic Post Hoc Approaches to False Positive Control.” <a href="https://hal.archives-ouvertes.fr/hal-02320543">https://hal.archives-ouvertes.fr/hal-02320543</a>.</p>
</div>
<div id="ref-tcga2014blca">
<p>Cancer Genome Atlas Research Network, and others. 2014. “Comprehensive Molecular Characterization of Urothelial Bladder Carcinoma.” <em>Nature</em> 507 (7492): 315.</p>
</div>
<div id="ref-CC2003">
<p>Cui, Xiangqin, and Gary A Churchill. 2003. “Statistical Tests for Differential Expression in cDNA Microarray Experiments.” <em>Genome Biol</em> 4 (4): 210.</p>
</div>
<div id="ref-GS2011">
<p>Goeman, Jelle J, and Aldo Solari. 2011. “Multiple Testing for Exploratory Research.” <em>Statistical Science</em> 26 (4): 584–97.</p>
</div>
<div id="ref-HSG2019">
<p>Hemerik, Jesse, Aldo Solari, and Jelle J Goeman. 2019. “Permutation-Based Simultaneous Confidence Bounds for the False Discovery Proportion.” <em>Biometrika</em> 106 (3): 635–49.</p>
</div>
<div id="ref-law2014voom">
<p>Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. “Voom: Precision Weights Unlock Linear Model Analysis Tools for Rna-Seq Read Counts.” <em>Genome Biology</em> 15 (2): 1–17.</p>
</div>
<div id="ref-limma">
<p>Ritchie, Matthew E, Belinda Phipson, Di Wu, Yifang Hu, Charity W Law, Wei Shi, and Gordon K Smyth. 2015. “limma Powers Differential Expression Analyses for RNA-Sequencing and Microarray Studies.” <em>Nucleic Acids Research</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007">https://doi.org/10.1093/nar/gkv007</a>.</p>
</div>
<div id="ref-simes86improved">
<p>Simes, R J. 1986. “An improved Bonferroni procedure for multiple tests of significance.” <em>Biometrika</em> 73 (3): 751–54.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.math.univ-toulouse.fr/~pneuvial/">Pierre Neuvial</a>, <a href="https://durandg12.github.io/">Guillermo Durand</a>, Nicolas Enjalbert-Courrech, <a href="https://marie-perrotdockes.github.io/">Marie Perrot-Dockès</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
